<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Trang T. Le" />
  <meta name="author" content="Daniel S. Himmelstein" />
  <meta name="author" content="Ariel A. Hippen Anderson" />
  <meta name="author" content="Matthew R. Gazzara" />
  <meta name="author" content="Casey S. Greene" />
  <meta name="dcterms.date" content="2020-05-03" />
  <meta name="keywords" content="computational biology, iscb, disparities" />
  <title>Analysis of ISCB honorees and keynotes reveals disparities</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/master/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Analysis of ISCB honorees and keynotes reveals disparities" />
  <meta name="citation_title" content="Analysis of ISCB honorees and keynotes reveals disparities" />
  <meta property="og:title" content="Analysis of ISCB honorees and keynotes reveals disparities" />
  <meta property="twitter:title" content="Analysis of ISCB honorees and keynotes reveals disparities" />
  <meta name="dc.date" content="2020-05-03" />
  <meta name="citation_publication_date" content="2020-05-03" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Trang T. Le" />
  <meta name="citation_author_institution" content="Department of Biostatistics, Epidemiology and Informatics, Institute for Biomedical Informatics, University of Pennsylvania" />
  <meta name="citation_author_orcid" content="0000-0003-3737-6565" />
  <meta name="twitter:creator" content="@trang1618" />
  <meta name="citation_author" content="Daniel S. Himmelstein" />
  <meta name="citation_author_institution" content="Department of Systems Pharmacology and Translational Therapeutics, Perelman School of Medicine, University of Pennsylvania" />
  <meta name="citation_author_orcid" content="0000-0002-3012-7446" />
  <meta name="twitter:creator" content="@dhimmel" />
  <meta name="citation_author" content="Ariel A. Hippen Anderson" />
  <meta name="citation_author_institution" content="Department of Systems Pharmacology and Translational Therapeutics, Perelman School of Medicine, University of Pennsylvania" />
  <meta name="citation_author_orcid" content="0000-0001-9336-6543" />
  <meta name="twitter:creator" content="@AHippenAnderson" />
  <meta name="citation_author" content="Matthew R. Gazzara" />
  <meta name="citation_author_institution" content="Department of Genetics, Perelman School of Medicine, University of Pennsylvania" />
  <meta name="citation_author_orcid" content="0000-0001-7710-4551" />
  <meta name="twitter:creator" content="@MR_Gazzara" />
  <meta name="citation_author" content="Casey S. Greene" />
  <meta name="citation_author_institution" content="Department of Systems Pharmacology and Translational Therapeutics, Perelman School of Medicine, University of Pennsylvania" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation" />
  <meta name="citation_author_orcid" content="0000-0001-8713-9213" />
  <meta name="twitter:creator" content="@greenescientist" />
  <link rel="canonical" href="https://greenelab.github.io/iscb-diversity-manuscript/" />
  <meta property="og:url" content="https://greenelab.github.io/iscb-diversity-manuscript/" />
  <meta property="twitter:url" content="https://greenelab.github.io/iscb-diversity-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://greenelab.github.io/iscb-diversity-manuscript/" />
  <meta name="citation_pdf_url" content="https://greenelab.github.io/iscb-diversity-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://greenelab.github.io/iscb-diversity-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://greenelab.github.io/iscb-diversity-manuscript/v/9500171d7d1a72cb69edd2ec8adb2264db210b8a/" />
  <meta name="manubot_html_url_versioned" content="https://greenelab.github.io/iscb-diversity-manuscript/v/9500171d7d1a72cb69edd2ec8adb2264db210b8a/" />
  <meta name="manubot_pdf_url_versioned" content="https://greenelab.github.io/iscb-diversity-manuscript/v/9500171d7d1a72cb69edd2ec8adb2264db210b8a/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <meta property="og:image" content="https://github.com/greenelab/iscb-diversity-manuscript/raw/9500171d7d1a72cb69edd2ec8adb2264db210b8a/build/assets/thumbnail.png" />
  <meta property="twitter:image" content="https://github.com/greenelab/iscb-diversity-manuscript/raw/9500171d7d1a72cb69edd2ec8adb2264db210b8a/build/assets/thumbnail.png" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Analysis of ISCB honorees and keynotes reveals disparities</h1>
</header>
<p><em>A DOI-citable version of this manuscript is available at <a href="https://doi.org/10.1101/2020.04.14.927251" class="uri">https://doi.org/10.1101/2020.04.14.927251</a></em>.</p>
<p>This version of the manuscript <a href="https://github.com/greenelab/iscb-diversity-manuscript/compare/v1.0...9500171d7d1a72cb69edd2ec8adb2264db210b8a">contains changes</a> subsequent to the <a href="https://github.com/greenelab/iscb-diversity-manuscript/releases/tag/v1.0">version 1.0 release</a>.</p>
<p><small><em>
This manuscript
(<a href="https://greenelab.github.io/iscb-diversity-manuscript/v/9500171d7d1a72cb69edd2ec8adb2264db210b8a/">permalink</a>)
was automatically generated
from <a href="https://github.com/greenelab/iscb-diversity-manuscript/tree/9500171d7d1a72cb69edd2ec8adb2264db210b8a">greenelab/iscb-diversity-manuscript@9500171</a>
on May 3, 2020.
</em></small></p>
<!-- include the Font Awesome library, per: https://fontawesome.com/start -->
<p><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css"></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Trang T. Le</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3737-6565">0000-0003-3737-6565</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/trang1618">trang1618</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/trang1618">trang1618</a><br>
<small>
Department of Biostatistics, Epidemiology and Informatics, Institute for Biomedical Informatics, University of Pennsylvania
</small></p></li>
<li><p><strong>Daniel S. Himmelstein</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3012-7446">0000-0002-3012-7446</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/dhimmel">dhimmel</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/dhimmel">dhimmel</a><br>
<small>
Department of Systems Pharmacology and Translational Therapeutics, Perelman School of Medicine, University of Pennsylvania
· Funded by the Gordon and Betty Moore Foundation (<a href="https://www.moore.org/grant-detail?grantId=GBMF4552">GBMF4552</a>)
</small></p></li>
<li><p><strong>Ariel A. Hippen Anderson</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9336-6543">0000-0001-9336-6543</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/arielah">arielah</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/AHippenAnderson">AHippenAnderson</a><br>
<small>
Department of Systems Pharmacology and Translational Therapeutics, Perelman School of Medicine, University of Pennsylvania
</small></p></li>
<li><p><strong>Matthew R. Gazzara</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7710-4551">0000-0001-7710-4551</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/mrgazzara">mrgazzara</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/MR_Gazzara">MR_Gazzara</a><br>
<small>
Department of Genetics, Perelman School of Medicine, University of Pennsylvania
</small></p></li>
<li><p><strong>Casey S. Greene</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8713-9213">0000-0001-8713-9213</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/cgreene">cgreene</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/greenescientist">greenescientist</a><br>
<small>
Department of Systems Pharmacology and Translational Therapeutics, Perelman School of Medicine, University of Pennsylvania; Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation
· Funded by the Gordon and Betty Moore Foundation (<a href="https://www.moore.org/grant-detail?grantId=GBMF4552">GBMF4552</a>)
</small></p></li>
</ul>
<h2 id="abstract">Abstract</h2>
<p>Delivering a keynote talk at a conference organized by a scientific society, or being named as a fellow by such a society, indicates that a scientist is held in high regard by their colleagues.
To explore if the distribution of such indicators of esteem in the field of bioinformatics reflects the composition of this field,
we compared the gender, country of affiliation, race/ethnicity and name-origin of 412 researchers who had been recognized by the <em>International Society for Computational Biology</em> (75 fellows and 337 keynote speakers) with 34,050 researchers who had been the corresponding authors on papers in three leading bioinformatics journals between 1993 and 2019.
The proportion of female fellows and keynote speakers was similar to that of the field overall.
However, fellows and keynote speakers with an affiliation in the United States were over-represented by a factor of 1.6;
moreover, almost half of this excess was accounted for by a deficit of 41 fellows and keynote authors from China, France and Italy.
Furthermore, within the US we found an excess of white fellows and keynote speakers, and a depletion of Asian fellows and keynote speakers.
Globally, names of East Asian origin have been persistently underrepresented among fellows and keynote speakers</p>
<h2 id="introduction">Introduction</h2>
<p>Scientists’ roles in society include identifying important topics of study, undertaking an investigation of those topics, and disseminating their findings broadly.
The scientific enterprise is largely self-governing: scientists act as peer reviewers on papers and grants, comprise hiring committees in academia, make tenure decisions, and select which applicants will be admitted to doctoral programs.
A lack of diversity in science could lead to pernicious biases that hamper the extent to which scientific findings are relevant to minority communities.
For example, a recent analysis found that minority scientists tend to apply for awards on topics with lower success rates <span class="citation" data-cites="hP9R3Quu">[<a href="#ref-hP9R3Quu" role="doc-biblioref">1</a>]</span>.
This finding might be the result of minority scientists selecting topics in more poorly funded areas.
Alternatively, reviewing scientists may not recognize the scientific importance of these topics, which may be of particular interest to minority scientists.
One way to address this issue is to directly examine peer recognition in different scientific fields.</p>
<p>Gender bias among conference speakers has been recognized as an area that can be improved with targeted interventions <span class="citation" data-cites="JHzGjxks w8Lq7zPE 1G16T65Sv 1Hmlj5vSu">[<a href="#ref-JHzGjxks" role="doc-biblioref">2</a>,<a href="#ref-w8Lq7zPE" role="doc-biblioref">3</a>,<a href="#ref-1G16T65Sv" role="doc-biblioref">4</a>,<a href="#ref-1Hmlj5vSu" role="doc-biblioref">5</a>]</span>.
Having more female organizers on conference committees is associated with having more female speakers <span class="citation" data-cites="UxTZTP0G">[<a href="#ref-UxTZTP0G" role="doc-biblioref">6</a>]</span>.
At medical conferences in the US and Canada, the proportion of female speakers is increasing at a modest rate <span class="citation" data-cites="6mHXFr1Z">[<a href="#ref-6mHXFr1Z" role="doc-biblioref">7</a>]</span>.
Gender bias appears to also influence funding decisions: an examination of scoring of proposals in Canada found that reviewers asked to assess the science produced a smaller gender gap in scoring than reviewers asked to assess the applicant <span class="citation" data-cites="19KEajiJN">[<a href="#ref-19KEajiJN" role="doc-biblioref">8</a>]</span>.
Challenges extend beyond gender: an analysis of awards at the NIH found that proposals by Asian, black or African-American applicants were less likely to be funded than those by white applicants <span class="citation" data-cites="njNa3CYa">[<a href="#ref-njNa3CYa" role="doc-biblioref">9</a>]</span>.
There are also potential interaction effects between gender and race or ethnicity that may particularly affect women of color’s efforts to gain NIH funding <span class="citation" data-cites="167tJVlJu">[<a href="#ref-167tJVlJu" role="doc-biblioref">10</a>]</span>.</p>
<p>We sought to understand the extent to which honors and high-profile speaking invitations were distributed equitably among gender, race/ethnicity, and name origin groups by an international society and its associated meetings.
As computational biologists, we focused on the <a href="https://www.iscb.org/">International Society for Computational Biology</a> (ISCB), its honorary Fellows as well as its affiliated international meetings that aim to have a global reach: <a href="https://en.wikipedia.org/wiki/Intelligent_Systems_for_Molecular_Biology">Intelligent Systems for Molecular Biology</a> (ISMB) and <a href="https://en.wikipedia.org/wiki/Research_in_Computational_Molecular_Biology">Research in Computational Molecular Biology</a> (RECOMB).</p>
<p>We used multiple methods to predict the gender, race/ethnicity, and name origins of honorees.
Existing methods were relatively US-centric because most of the data was derived in whole or in part from the US Census.
We scraped more than 700,000 entries from English-language Wikipedia that contained nationality information to complement these existing methods and built multiple machine learning classifiers to predict name origin.
We also examined the last and corresponding authors for publications in ISCB partner journals to establish a field-specific baseline using the same metrics.
The results were consistent across all approaches: we found a dearth of non-white speakers and honorees.
The lack of Asian scientists among keynote speakers and Fellows was particularly pronounced when compared against the field-specific background.</p>
<h2 id="results">Results</h2>
<h3 id="curated-honorees-and-literature-derived-potential-honorees">Curated Honorees and Literature-derived Potential Honorees</h3>
<p>We curated a dataset of ISCB honorees that included 412 honorees who were keynote speakers at international ISCB-associated conferences (ISMB and RECOMB) as well as ISCB Fellows.
The ISCB Fellows set contained the complete set of Fellows named (2009–2019).
Keynote speakers were available for ISMB for all years from 1993–2019.
Keynote speakers for RECOMB were available for all years from 1997–2019.
We included individuals who were honored multiple times as separate entries.
For example, Christine Orengo was a keynote speaker at RECOMB 2004 and became an ISCB Fellow in 2016, and thus was counted twice in this list.</p>
<p>We sought to compare this dataset with a background distribution of potential speakers, which we considered to be last or senior authors of bioinformatics and computational biology manuscripts.
We used those published in <a href="https://academic.oup.com/bioinformatics">Bioinformatics</a>, <a href="https://bmcbioinformatics.biomedcentral.com/">BMC Bioinformatics</a>, and <a href="https://journals.plos.org/ploscompbiol/">PLOS Computational Biology</a> as a set of bioinformatics and computational biology manuscripts.
We downloaded the metadata of manuscripts published in these journals from PubMed, which provided almost 30,000 articles for evaluation.
However, although PubMed provides author order, it does not provide corresponding author information.
To determine corresponding authors for an article, we used the PMC corresponding author information when it was available (20,696 articles) and the PubMed last author as a fallback when corresponding author information was missing (9,053 articles).</p>
<h3 id="assessing-gender-diversity-of-authors-and-honorees">Assessing Gender Diversity of Authors and Honorees</h3>
<p>Although <em>Bioinformatics</em> was established in 1998 and <em>BMC Bioinformatics</em> in 2000, the metadata for these journal papers before 2002 only have initials for first and/or middle author names.
Therefore, without first and middle names, we do not have author gender predictions before this year.</p>
<p>We observed a slow increase of the proportion of predicted female authors, arriving at just over 20% in 2019 (Fig. <a href="#fig:gender_breakdown">1</a>, left).
We observe very similar trend within each journal, but estimated female proportion has increased the most in <em>BMC Bioinformatics</em> (see <a href="https://greenelab.github.io/iscb-diversity/10.visualize-gender.html#sup_fig_s1">notebook</a>).
In recent years, ISCB Fellows and keynote speakers appear to be more evenly split between men and women compared to the population of authors published in computational biology and bioinformatics journals (Fig. <a href="#fig:gender_breakdown">1</a>, right); however, it has not yet reached parity.
Further, taking all the years together, a Welch two-sample t-test did not reveal any statistically significant difference in the mean probability of ISCB honorees predicted to be female compared to that of authors (t<sub>418</sub> = 0.753, <em>p</em> = 0.226).
We observed an increasing trend of honorees who were women in each honor category, especially in the group of ISCB Fellows (see <a href="https://greenelab.github.io/iscb-diversity/10.visualize-gender.html#sup_fig_s1">notebook</a>), which markedly increased after 2015.
Through 2019, there were a number of years when meetings or ISCB Fellow classes have a high probability of recognizing only male honorees and none that appeared to have exclusively female honorees.</p>
<div id="fig:gender_breakdown" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/greenelab/iscb-diversity/3853a20773764cedcf6aa61f38f062b26bd7655d/figs/gender_breakdown.png" style="width:70.0%" alt="" /><figcaption><span>Figure 1:</span> ISCB Fellows and keynote speakers appear more evenly split between men and women than PubMed authors in recent years, but the proportion has not reached parity. Estimated composition of gender prediction over the years of
all Pubmed computational biology and bioinformatics journal authors (left),
and all ISCB Fellows and keynote speakers (right)
was computed as the average of prediction probabilities of Pubmed articles or ISCB honorees each year.</figcaption>
</figure>
</div>
<h3 id="predicting-name-origin-groups-with-lstm-neural-networks-and-wikipedia">Predicting Name Origin Groups with LSTM Neural Networks and Wikipedia</h3>
<p>Table <a href="#tbl:example_names">1</a> shows the size of the training set for each of the name origin groups as well as a few examples of PubMed author names that had at least 90% prediction probability in that group.
We refer to this dataset as Wiki2019 (available online in <a href="https://github.com/greenelab/wiki-nationality-estimate/blob/master/data/annotated_names.tsv"><code>annotated_names.tsv</code></a>).</p>
<div id="tbl:example_names" class="tablenos">
<table>
<caption><span>Table 1:</span> <strong>Predicting name-origin groups of names trained on Wikipedia’s living people.</strong>
The table lists the 10 groups and the number of living people for each region that the LSTM was trained on.
Example names shows actual author names that received a high prediction for each region.
Full information about which countries comprised each region can be found in the online dataset <a href="https://github.com/greenelab/iscb-diversity/blob/make-letters/data/countries/2020-01-31_groupings.tsv"><code>country_to_region.tsv</code></a>.
</caption>
<colgroup>
<col style="width: 18%" />
<col style="width: 18%" />
<col style="width: 62%" />
</colgroup>
<thead>
<tr class="header">
<th>Group</th>
<th>Training Size</th>
<th>Example Names</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Celtic/English names</td>
<td>154,890</td>
<td>Adam O Hebb, Oliver G Pybus, David W Ritchie, James WJ Anderson, James W MacDonald, Robert Clarke</td>
</tr>
<tr class="even">
<td>European names</td>
<td>78,157</td>
<td>Tracey M Filzen, Jos H Beijnen, Caroline Louis-Jeune, Christian Lorenzi, Boris Vassilev, Verena Heinrich</td>
</tr>
<tr class="odd">
<td>Hispanic names</td>
<td>66,931</td>
<td>Ramón Latorre, Antonio J Jimeno-Yepes, Felipe A Simão, Paulo S L de Oliveira, Juan Carlos Rodríguez-Manzaneque, Natalia Acevedo-Luna</td>
</tr>
<tr class="even">
<td>East Asian names</td>
<td>54,588</td>
<td>Heejoon Chae, Wenchao Jiang, Haizhou Liu, Miho Uchida, Wenxuan Zhang, Jiali Feng</td>
</tr>
<tr class="odd">
<td>Arabic names</td>
<td>31,418</td>
<td>Hamidreza Chitsaz, Farzad Sangi, Habib Motieghader, Berke Ç Toptas, Ali Aliyari, Bülent Arman Aksoy</td>
</tr>
<tr class="even">
<td>Nordic names</td>
<td>28,978</td>
<td>Cecilia M Lindgren, Ellen Larsen, Jesper R Gådin, Janne H Korhonen, Johan Åqvist, Jens Nilsson</td>
</tr>
<tr class="odd">
<td>South Asian names</td>
<td>20,025</td>
<td>Amitabh Chak, Matthew G Seetin, Matrika Gupta, Sumudu P Leelananda, VS Kumar Kolli, Swanand Gore</td>
</tr>
<tr class="even">
<td>African names</td>
<td>17,826</td>
<td>Timothy Kinyanjui, Jammbe Z Musoro, Nyaradzo M Mgodi, Magambo Phillip Kimuda, Probhonjon Baruah, Adaoha E C Ihekwaba</td>
</tr>
<tr class="odd">
<td>Hebrew names</td>
<td>4,549</td>
<td>Alexander J Sadovsky, Boris Shraiman, Gil Goldshlager, Eytan Adar, Aviva Peleg, Nir Esterman</td>
</tr>
<tr class="even">
<td>Greek names</td>
<td>4,138</td>
<td>Gianni Panagiotou, Themis Lazaridis, Eleni Mijalis, Nikolaos Tsiantis, Konstantinos A Kyritsis, Dimitris E Messinis</td>
</tr>
</tbody>
</table>
</div>
<p>We next aimed to predict the name origin groups of honorees and authors.
We constructed a training dataset with more than 700,000 name-nationality pairs by parsing the English-language Wikipedia.
We trained a LSTM neural network on n-grams to predict name groups.
We found similar performance across 1, 2, and 3-grams;
however, because the classifier required fewer epochs to train with 3-grams, we used this length in the model that we term Wiki2019-LSTM.
Our Wiki2019-LSTM returns, for each given name, a probability of that name originating from each of the specified 10 groups.
We observed a multiclass area under the receiver operating characteristic curve (AUC) score of 95.9% for the classifier, indicating that the classifier can recapitulate name origins with high sensitivity and specificity.
For each individual group, the high AUC (above 92%, Fig. <a href="#fig:wiki2019_lstm">2</a>A) suggests that our classifier was sufficient for use in a broad-scale examination of disparities.
We also observed that the model was well calibrated (Fig. <a href="#fig:wiki2019_lstm">2</a>B).
We also examined potential systematic errors between pairs of name origin groupings with a confusion heatmap and did not find off-diagonal enrichment for any pairing (Fig. <a href="#fig:wiki2019_lstm">2</a>C).</p>
<div id="fig:wiki2019_lstm" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/greenelab/iscb-diversity/3853a20773764cedcf6aa61f38f062b26bd7655d/figs/fig_3.png" style="width:100.0%" alt="" /><figcaption><span>Figure 2:</span> The Wiki2019-LSTM model performs well on the testing dataset.
The area under the ROC curve is above 92% for each category, showing strong performance across origin categories (A).
A calibration curve, computed with the caret R package, shows consistency between the predicted probabilities (midpoints of each fixed-width bin) and the observed fraction of names in each bin (B).
Heatmap showing whether names from a given group (x-axis) received higher (purple) or lower (green) predictions for each group (y-axis) than would be expected by group prevalence alone (C).
The values represent log<sub>2</sub> fold change between the average predicted probability and the prevalence of the corresponding predicted group in the testing dataset (null).
Scaling by group prevalence accounts for the imbalance of groups in the testing dataset.
In all cases, the classifier predicts the true groups above the expected null probability (matrix diagonals are all purple).
For off-diagonal cells, darker green indicates a lower mean prediction compared to the null.
For example, the classifier does not often mistake East Asian names as Greek, but is more prone to mistaking South Asian names as Celtic/English.</figcaption>
</figure>
</div>
<h3 id="assessing-the-name-origin-diversity-of-authors-and-honorees">Assessing the Name Origin Diversity of Authors and Honorees</h3>
<p>We applied our Wiki2019-LSTM model to both our computational biology honorees dataset and our dataset of corresponding authors.
We found that the proportion of authors with Celtic/English names had decreased (Fig. <a href="#fig:region_breakdown">3</a>A, left), particularly for papers published in <em>Bioinformatics</em> and <em>BMC Bioinformatics</em> (see <a href="https://greenelab.github.io/iscb-diversity/11.visualize-name-origins.html#sup_fig_s4">notebook</a>).
Among keynote speakers and fellows, we found that the majority of honorees are predicted to have Celtic/English or European names (Fig. <a href="#fig:region_breakdown">3</a>A, right).
When we directly compared honoree composition with PubMed, we observed discrepancies between the two groups, namely a large overrepresentation of keynote speakers with Celtic/English names (t<sub>418</sub> = 9.01, <em>p</em> &lt; 10<sup>-16</sup>) and a substantial underrepresentation of keynote speakers with East Asian names (t<sub>454</sub> = -15.8, <em>p</em> &lt; 10<sup>-16</sup>) (Fig. <a href="#fig:region_breakdown">3</a>B).
No statistically significant difference was observed between the proportion of honorees and authors with European names (t<sub>425</sub> = 0.0282, <em>p</em> = 0.489) or in other categories (see Table <a href="#tbl:example_names">1</a>, t<sub>425</sub> = -0.176, <em>p</em> = 0.860).</p>
<div id="fig:region_breakdown" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/greenelab/iscb-diversity/3853a20773764cedcf6aa61f38f062b26bd7655d/figs/region_breakdown.png" alt="" /><figcaption><span>Figure 3:</span> Compared to the name collection of Pubmed authors, honorees with Celtic/English names are overrepresented while honorees with East Asian names are underrepresented.
No statistically significant difference was observed between the proportion of honorees and authors with European names or in other categories (see Table <a href="#tbl:example_names">1</a>).
Estimated composition of name origin prediction over the years of
(A, left) all Pubmed computational biology and bioinformatics journal authors,
and (A, right) all ISCB Fellows and keynote speakers
was computed as the average of prediction probabilities of Pubmed articles or ISCB honorees each year.
(B) For each region, the mean predicted probability of Pubmed articles is shown as teal LOESS curve, and the mean probability and 95% confidence interval of the ISCB honoree predictions are shown as dark circles and vertical lines.</figcaption>
</figure>
</div>
<h3 id="affiliation-analysis">Affiliation Analysis</h3>
<p>We analyzed the countries of affiliation between corresponding authors and ISCB honorees.
For each country, we report a value of log<sub>2</sub> enrichment (LOE) and its 95% confidence intervals (Table <a href="#tbl:country-enrichment">2</a>).
A positive value of LOE indicates a higher proportion of honorees affiliated with that country compared to authors.
A LOE value of 1 represents a one-fold enrichment (i.e., observed number of honorees is twice as much as expected).
In the 20 countries with the most publications, we found an overrepresentation of honorees affiliated with institutions and companies in the US (97 speakers more than expected, LOE = 0.6, 95% CI (0.5, 0.8)) and Israel (12 speakers more than expected, LOR = 1.6, 95% CI (0.9, 2.3)), and an underrepresentation of honorees affiliated with those in China, France, Italy, the Netherlands, Taiwan, and India (Fig. <a href="#fig:country-enrichment">4</a>).</p>
<div id="fig:country-enrichment" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/greenelab/iscb-diversity/3853a20773764cedcf6aa61f38f062b26bd7655d/figs/enrichment-plot.png" style="width:80.0%" alt="" /><figcaption><span>Figure 4:</span> The overrepresentation of honorees affiliated with institutions and companies in the US and Israel contrasts the underrepresentation of honorees affiliated with those in China, France, Italy, the Netherlands, Taiwan, and India.
Each country’s log<sub>2</sub> enrichment (LOE) and its 95% confidence interval are displayed on the left.
Observed (triangle) and expected (circle) number of honorees and their differences (observed - expected) are shown in square-root scale on the right.
Countries are ordered based on the proportion of authors in the field.</figcaption>
</figure>
</div>
<div id="tbl:country-enrichment" class="tablenos">
<table style="width:100%;">
<caption><span>Table 2:</span> <strong>Enrichment and depletion in proportion of ISCB honorees compared to Pubmed corresponding authors of 20 countries with the most publications.</strong>
The table lists the countries and their corresponding enrichment, which we computed by dividing the observed proportion of honorees by expected proportion of honorees.
The expected proportion was calculated using corresponding author proportions.
A positive Log<sub>2</sub>(Enrichment) indicated a higher proportion of honorees than corresponding authors affiliated with that country.
The full table with all countries can be browsed interactively in the corresponding <a href="https://greenelab.github.io/iscb-diversity/12.analyze-affiliation.html#enrichment_tab">analysis notebook</a>.
</caption>
<colgroup>
<col style="width: 12%" />
<col style="width: 14%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 16%" />
<col style="width: 9%" />
<col style="width: 13%" />
<col style="width: 19%" />
</colgroup>
<thead>
<tr class="header">
<th>Country</th>
<th>Author proportion</th>
<th>Observed</th>
<th>Expected</th>
<th>Observed - Expected</th>
<th>Enrichment</th>
<th>Log<sub>2</sub>(Enrichment)</th>
<th>95% Confidence Interval</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>United States</td>
<td>38.76%</td>
<td>237.5</td>
<td>152.7</td>
<td>84.8</td>
<td>1.6</td>
<td>0.6</td>
<td>(0.4, 0.8)</td>
</tr>
<tr class="even">
<td>United Kingdom</td>
<td>8.36%</td>
<td>36.0</td>
<td>32.9</td>
<td>3.1</td>
<td>1.1</td>
<td>0.1</td>
<td>(-0.4, 0.6)</td>
</tr>
<tr class="odd">
<td>Germany</td>
<td>7.55%</td>
<td>27.0</td>
<td>29.7</td>
<td>-2.7</td>
<td>0.9</td>
<td>-0.1</td>
<td>(-0.7, 0.4)</td>
</tr>
<tr class="even">
<td>China</td>
<td>5.82%</td>
<td>3.0</td>
<td>22.9</td>
<td>-19.9</td>
<td>0.1</td>
<td>-2.9</td>
<td>(-5.2, -1.4)</td>
</tr>
<tr class="odd">
<td>France</td>
<td>3.86%</td>
<td>4.0</td>
<td>15.2</td>
<td>-11.2</td>
<td>0.3</td>
<td>-1.9</td>
<td>(-3.8, -0.6)</td>
</tr>
<tr class="even">
<td>Italy</td>
<td>3.04%</td>
<td>2.0</td>
<td>12.0</td>
<td>-10.0</td>
<td>0.2</td>
<td>-2.6</td>
<td>(-5.6, -0.7)</td>
</tr>
<tr class="odd">
<td>Canada</td>
<td>3.03%</td>
<td>12.0</td>
<td>11.9</td>
<td>0.1</td>
<td>1.0</td>
<td>0.0</td>
<td>(-1, 0.8)</td>
</tr>
<tr class="even">
<td>Japan</td>
<td>2.44%</td>
<td>9.0</td>
<td>9.6</td>
<td>-0.6</td>
<td>0.9</td>
<td>-0.1</td>
<td>(-1.2, 0.8)</td>
</tr>
<tr class="odd">
<td>Spain</td>
<td>2.39%</td>
<td>6.0</td>
<td>9.4</td>
<td>-3.4</td>
<td>0.6</td>
<td>-0.7</td>
<td>(-2.1, 0.5)</td>
</tr>
<tr class="even">
<td>Australia</td>
<td>2.33%</td>
<td>5.0</td>
<td>9.2</td>
<td>-4.2</td>
<td>0.5</td>
<td>-0.9</td>
<td>(-2.5, 0.4)</td>
</tr>
<tr class="odd">
<td>Netherlands</td>
<td>1.91%</td>
<td>1.0</td>
<td>7.5</td>
<td>-6.5</td>
<td>0.1</td>
<td>-2.9</td>
<td>(-8.2, -0.4)</td>
</tr>
<tr class="even">
<td>Switzerland</td>
<td>1.81%</td>
<td>7.0</td>
<td>7.1</td>
<td>-0.1</td>
<td>1.0</td>
<td>-0.0</td>
<td>(-1.4, 1)</td>
</tr>
<tr class="odd">
<td>Israel</td>
<td>1.46%</td>
<td>17.5</td>
<td>5.8</td>
<td>11.7</td>
<td>3.0</td>
<td>1.6</td>
<td>(0.8, 2.3)</td>
</tr>
<tr class="even">
<td>Sweden</td>
<td>1.34%</td>
<td>6.0</td>
<td>5.3</td>
<td>0.7</td>
<td>1.1</td>
<td>0.2</td>
<td>(-1.3, 1.3)</td>
</tr>
<tr class="odd">
<td>Korea</td>
<td>1.30%</td>
<td>1.0</td>
<td>5.1</td>
<td>-4.1</td>
<td>0.2</td>
<td>-2.4</td>
<td>(-7.7, 0.1)</td>
</tr>
<tr class="even">
<td>Taiwan</td>
<td>1.25%</td>
<td>0.0</td>
<td>4.9</td>
<td>-4.9</td>
<td>0.0</td>
<td>-Inf</td>
<td>(-Inf, -0.4)</td>
</tr>
<tr class="odd">
<td>India</td>
<td>1.20%</td>
<td>0.0</td>
<td>4.7</td>
<td>-4.7</td>
<td>0.0</td>
<td>-Inf</td>
<td>(-Inf, -0.3)</td>
</tr>
<tr class="even">
<td>Belgium</td>
<td>1.04%</td>
<td>1.0</td>
<td>4.1</td>
<td>-3.1</td>
<td>0.2</td>
<td>-2.0</td>
<td>(-7.3, 0.5)</td>
</tr>
<tr class="odd">
<td>Singapore</td>
<td>0.88%</td>
<td>1.0</td>
<td>3.5</td>
<td>-2.5</td>
<td>0.3</td>
<td>-1.8</td>
<td>(-7.1, 0.7)</td>
</tr>
<tr class="even">
<td>Finland</td>
<td>0.85%</td>
<td>0.0</td>
<td>3.4</td>
<td>-3.4</td>
<td>0.0</td>
<td>-Inf</td>
<td>(-Inf, 0.1)</td>
</tr>
</tbody>
</table>
</div>
<h3 id="assessing-the-racial-and-ethnic-diversity-of-us-affiliated-authors-and-honorees">Assessing the Racial and Ethnic Diversity of US-affiliated Authors and Honorees</h3>
<p>We predicted the race and ethnicity of US-affiliated authors and honorees using wru, which is based on US census data.
We found that an increasing proportion of authors in computational biology and bioinformatics journals had last names associated with selecting Asian as a race/ethnicity category in the US census (Fig. <a href="#fig:us_racial_makeup">5</a>A).
This was primarily driven by publications in <em>Bioinformatics</em> and <em>BMC Bioinformatics</em> (Fig. <a href="#fig:us_racial_makeup">5</a>B, top).
We did not observe a corresponding increase at <em>PLOS Computational Biology</em> (Fig. <a href="#fig:us_racial_makeup">5</a>B, bottom).
Compared to Pubmed authors, ISCB honorees with US affiliations have a higher proportion of individuals whose last names we associated with selecting white as a race/ethnicity category in the US census (Fig. <a href="#fig:us_racial_makeup">5</a>C vs. A).
Separating honoree results by honor category did not reveal any clear differences (Fig. <a href="#fig:us_racial_makeup">5</a>D).</p>
<div id="fig:us_racial_makeup" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/greenelab/iscb-diversity/3853a20773764cedcf6aa61f38f062b26bd7655d/figs/us_racial_makeup.png" alt="" /><figcaption><span>Figure 5:</span> We find an overrepresentation of white and underrepresentation of Asian honorees as compared to authors. Estimated composition of census-based race/ethnicity prediction over the years of
(A) all Pubmed computational biology and bioinformatics journal authors,
(B) authors in each journal,
(C) all ISCB Fellows and keynote speakers,
and (D) ISCB honorees in each honor category
was computed as the average of prediction probabilities of Pubmed articles or ISCB honorees each year.
For each race/ethnicity category, the mean predicted probability of Pubmed articles is shown as teal LOESS curve, and the mean probability and 95% confidence interval of the ISCB honoree predictions are shown as dark circles and vertical lines (E).</figcaption>
</figure>
</div>
<p>We directly compared honoree and author results from 1993 to 2019 for the predicted proportion of white, Asian, and other categories (Fig. <a href="#fig:us_racial_makeup">5</a>E).
We found that, over the years, white honorees have been significantly overrepresented (t<sub>218</sub> = 14.8, <em>p</em> &lt; 10<sup>-16</sup>) and Asian honorees have been significantly underrepresented (t<sub>236</sub> = -18.8, <em>p</em> &lt; 10<sup>-16</sup>).
A Welch two-sample t-test did not reveal any statistically significant difference in the mean probability of ISCB speakers predicted to be in Other categories compared to authors (t<sub>203</sub> = 1.79, <em>p</em> = 0.0747).</p>
<h3 id="assessing-the-name-origin-diversity-of-us-affiliated-authors-and-honorees">Assessing the Name Origin Diversity of US-affiliated Authors and Honorees</h3>
<p>We applied our Wiki2019-LSTM model to US-affiliated authors and honorees and found similar pattern as when we considered all authors and honorees.
We note that the US was not included in the training of the Wiki2019-LSTM model (see Methods).
Specifically, the proportion of authors with Celtic/English names had decreased (Fig. <a href="#fig:us_name_origin">6</a>A, left), particularly for papers published in <em>Bioinformatics</em> and <em>BMC Bioinformatics</em> (see <a href="https://greenelab.github.io/iscb-diversity/14.us-name-origin.html#sup_fig_s6">notebook</a>).
Meanwhile, the majority of honorees are predicted to have Celtic/English or European names (Fig. <a href="#fig:us_name_origin">6</a>A, right).
When we directly compared honoree composition with PubMed, we observed discrepancies between the two groups, namely a large overrepresentation of honorees with Celtic/English names (t<sub>247</sub> = 6.46, <em>p</em> &lt; 10<sup>-9</sup>), a smaller overrepresentation of honorees with European names (t<sub>250</sub> = 2.54, <em>p</em> = 0.0059), and a substantial underrepresentation of honorees with East Asian names (t<sub>329</sub> = -20.6, <em>p</em> &lt; 10<sup>-16</sup>) (Fig. <a href="#fig:us_name_origin">6</a>B).</p>
<div id="fig:us_name_origin" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/greenelab/iscb-diversity/3853a20773764cedcf6aa61f38f062b26bd7655d/figs/us_name_origin.png" alt="" /><figcaption><span>Figure 6:</span> Compared to the name collection of US-affiliated Pubmed authors, US-affiliated honorees with Celtic/English and European names are overrepresented while US-affiliated honorees with East Asian names are underrepresented.
No statistically significant difference was observed between the proportion of honorees and authors with names in other categories (see Table <a href="#tbl:example_names">1</a>, t<sub>252</sub> = -0.152, <em>p</em> = 0.879).
Estimated composition of name origin prediction over the years of
(A, left) US-affiliated journal authors,
and (A, right) all US-affiliated Fellows and keynote speakers
was computed as the average of prediction probabilities of US-affiliated corresponding authors or ISCB honorees each year.
(B) For each region, the mean predicted probability of US-affiliated corresponding authors is shown as teal LOESS curve, and the mean probability and 95% confidence interval of the US-affiliated ISCB honoree predictions are shown as dark circles and vertical lines.</figcaption>
</figure>
</div>
<h3 id="reception-feedback">Reception &amp; Feedback</h3>
<p>We released <a href="https://github.com/greenelab/iscb-diversity-manuscript/releases/tag/v1.0">version 1.0</a> of our manuscript on 2020-01-30.
We submitted this version to the 28<sup>th</sup> annual ISMB conference planned for July 2020 in Montreal.
The study received public feedback almost immediately.
Specifically, a community member <a href="https://github.com/greenelab/iscb-diversity-manuscript/issues/27">pointed out</a> problems with the country-group names we imported from NamePrism.</p>
<p>On 2020-02-20, ISCB released a <a href="https://www.iscb.org/iscb-news-items/4257-2020-feb20-iscb-announcement">statement on equity and inclusion</a> that alluded to our study, but did not directly name or link to it.
The statement contained the following paragraph:</p>
<blockquote>
<p>We continually revisit our procedures for progress towards equity and inclusion and assess their efficacy.
We do not think that automated procedures of assessment are appropriate for this purpose.
Algorithms performing such categorization suffer from inherent limitations that distort the notion of balance we are striving for.
Specifically, categorization of disjoint regional, ethnic or gender groups of scientists, while easing classification, also functions divisively.</p>
</blockquote>
<p>Our ISMB submission was reviewed by 5 peer reviewers and <a href="https://github.com/greenelab/iscb-diversity-manuscript/issues/37">rejected</a> on 2020-03-13.
We thank the reviewers for their input, and have made our <a href="https://github.com/greenelab/iscb-diversity-manuscript/blob/master/content/response-to-reviewers.md">responses public</a>).</p>
<p>Much of the feedback centered around the inherent imperfections of automated detection of race or nationality.
Therefore, we expanded our assessment to include country of affiliation, which can be detected with high fidelity from PubMed metadata.
Adding affiliation information allowed us to examine disparities within a single country and to rule out geographic disparities as the sole factor leading to an underrepresentation of Asian honorees.</p>
<p>We appreciate the feedback we have received.
Early indications suggest ISCB may be heeding the findings of our analysis and increasing the diversity of honorees.
In 2020, among 12 ISCB Fellows and 5 ISMB keynote speakers, the mean predicted probability of each honoree having an East Asian name is 33%, higher than any estimate in previous years (see <a href="https://greenelab.github.io/iscb-diversity/15.analyze-2020.html">notebook</a>).
The set of honorees also includes the first ISCB Fellow from China.
Compared to past years, the 2020 honorees appear to better reflect the diversity of scientists in the computational biology field.</p>
<h2 id="discussion">Discussion</h2>
<p>A major challenge that we faced in carrying out this work was to narrow down geographic origins for some groups of names.
Some name origin groups, such as Hispanic names, are geographically disparate.
We were unable to construct a classifier that could distinguish between names from Iberian countries (Spain and Portugal) from those in Latin America in the group of Hispanic names.
Discrepancies in representation between these groups are thus undetectable by our classifier.
Honoree counts of those with Hispanic names are influenced from Spain as well as Latin America.
In such cases, our analyses may substantially understate the extent to which minoritized scientists are underrepresented among honorees and authors.</p>
<p>Biases in authorship practices may also result in our underestimation of the composition of minoritized scientists within the field.
We estimated the composition of the field using corresponding author status, but in neuroscience <span class="citation" data-cites="14PjsfIPd">[<a href="#ref-14PjsfIPd" role="doc-biblioref">11</a>]</span> and other disciplines <span class="citation" data-cites="17aOPYsbT">[<a href="#ref-17aOPYsbT" role="doc-biblioref">12</a>]</span> women are underrepresented among such authors.
Such an effect would cause us to underestimate the number of women in the field.
Though this effect has been studied with respect to gender, we are not aware of similar work examining race, ethnicity, or name origins.</p>
<p>We acknowledged that our supervised learning approaches are neither error free nor bias free.
Because wru was trained on the US census to make predictions, many of the missing predictions are on names not observed in the US census.
Although the underestimation of the proportion of these names could not be quantified in the list of authors and honorees, we complemented this race/ethnicity prediction method with an additional name origin analysis.
By integrating different methods and preserving uncertainty by analyzing prediction probabilities rather than applying a hard assignment for each prediction, we hope to alleviate these biases and discover insightful findings that correctly reflect the current representation diversity at conferences.</p>
<p>Focusing on an international society and meetings, we measured honor and authorship rates worldwide.
In this setting, we observe disparities by name groups.
Because invitation and honor patterns could be driven by biases associated with name groups, geography, or other factors, we cross-referenced name group predictions with author affiliations to disentangle the relationship between geographic regions, name groups and invitation probabilities.
We found that disparities persisted even within the group of honorees with a US affiliation.</p>
<p>An important questions to ask when measuring representation is what the right level of representation is.
We suggest that considering equity may be more appropriate than strictly diversity.
For example, we found similar representation of women between authors and honorees, which suggests honoree diversity is similar to that of authors.
However, if fewer women are in the field because of systemic factors that inhibit their participation, we would not have reached equity.
In addition to holding fewer corresponding authorship positions, on average, female scientists of different disciplines are cited less often <span class="citation" data-cites="9TNbcewU">[<a href="#ref-9TNbcewU" role="doc-biblioref">13</a>]</span>, invited by journals to submit papers less often <span class="citation" data-cites="17aOPYsbT">[<a href="#ref-17aOPYsbT" role="doc-biblioref">12</a>]</span>, suggested as reviewers less often <span class="citation" data-cites="1EYk4f8Mu">[<a href="#ref-1EYk4f8Mu" role="doc-biblioref">15</a>]</span>, and receive significantly worse review scores <span class="citation" data-cites="5HWeup6K">[<a href="#ref-5HWeup6K" role="doc-biblioref">14</a>]</span>.
Societies, both through their honorees and the individuals who deliver keynotes at their meetings, can play a positive role in improving the presence of female STEM role models, which, for example, may lead to higher persistence for undergraduate women in geoscience <span class="citation" data-cites="1FX4fLEIy">[<a href="#ref-1FX4fLEIy" role="doc-biblioref">16</a>]</span>.
Efforts are underway to create Wikipedia entries for more female <span class="citation" data-cites="zdS2Wmy2">[<a href="#ref-zdS2Wmy2" role="doc-biblioref">17</a>]</span> and black, Asian, and minority scientists <span class="citation" data-cites="yaljvCUU">[<a href="#ref-yaljvCUU" role="doc-biblioref">18</a>]</span>, which can help early-career scientists identify role models.
We find that ISCB’s honorees and keynote speakers, though not yet reaching gender parity, appear to have similar gender proportion to the field as a whole.
On the other hand, honorees include significantly fewer people of color than the field as a whole, and Asian scientists are dramatically underrepresented among honorees.
Although we estimate the fraction of non-white and non-Asian authors to be relatively similar to the estimated honoree rate, we note that both are represented at levels substantially lower than in the US population.
Societies can play a positive role in enhancing equity if they design policies to honor scientists in ways that counter these biases.</p>
<p>The central role that scientists play in evaluating each other and each other’s findings makes equity critical.
Even many nominally objective methods of assessing excellence (e.g., h-index, grant funding obtained, number of high-impact peer-reviewed publications, and total number of peer-reviewed publications) are subject to the bias of peers during review.
These could be affected by explicit biases, implicit biases, or pernicious biases in which a reviewer might consider a path of inquiry, as opposed to an individual, to be more or less meritorious based on the reviewer’s own background <span class="citation" data-cites="hP9R3Quu">[<a href="#ref-hP9R3Quu" role="doc-biblioref">1</a>]</span>.
Our efforts to measure the diversity of honorees in an international society suggests that, while a focus on gender parity may be improving some aspects of diversity among honorees, contributions from scientists of color are underrecognized.</p>
<h2 id="materials-and-methods">Materials and Methods</h2>
<h3 id="honoree-curation">Honoree Curation</h3>
<p>From <a href="http://web.archive.org/web/20200116150052/https://www.iscb.org/iscb-fellows">ISCB’s webpage listing <strong>ISCB Distinguished Fellows</strong></a>, we found recipients listed by their full names for the years 2009–2019.
We gleaned the full name of the Fellow as well as the year in which they received the honor.
We identified major ISCB-associated conferences as those designated flagship (ISMB) or those that had been held on many continents (RECOMB).
To identify <strong>ISMB Keynote Speakers</strong>, we examined the webpage for each ISMB meeting.
The invited speakers at ISMB before 2001 were listed in the Preface pages of each year’s proceedings, which were archived in the <a href="https://aaai.org/Library/ISMB/ismb-library.php">ISMB collection</a> of the AAAI digital library.
We found full names of all keynote speakers for the years 1993–2019.</p>
<p>For the RECOMB meeting, we found conference webpages with keynote speakers for 1999, 2000, 2001, 2004, 2007, 2008, and 2010–2019.
We were able to fill in the missing years using information from the RECOMB 2016 proceedings, which summarizes the first 20 years of the RECOMB conference <span class="citation" data-cites="1Bji9UOxF">[<a href="#ref-1Bji9UOxF" role="doc-biblioref">19</a>]</span>.
This volume has two tables of keynote speakers from 1997–2006 (Table 14, page XXVII) and 2007–2016 (Table 4, page 8).
Using these tables to verify the conference speaker lists, we arrived at two special instances of inclusion/exclusion.
Although Jun Wang was not included in these tables, we were able to confirm that he was a keynote speaker in 2011 with the RECOMB 2011 proceedings <span class="citation" data-cites="DBJsvxt7">[<a href="#ref-DBJsvxt7" role="doc-biblioref">20</a>]</span>, and thus we included this speaker in the dataset.
Marian Walhout was invited as a keynote speaker but had to <a href="http://recomb2015.mimuw.edu.pl/node/18.html">cancel</a> the talk due to other obligations.
Because her name was neither mentioned in the 2015 proceedings <span class="citation" data-cites="aziMb8CD">[<a href="#ref-aziMb8CD" role="doc-biblioref">21</a>]</span> nor in the above-mentioned tables, we excluded this speaker from our dataset.</p>
<h4 id="name-processing">Name processing</h4>
<p>When extracting honoree names, we began with the full name as provided on the site.
Because our prediction methods required separated first and last names, we chose the first non-initial name as the first name and the final name as the last name.
We did not consider a hyphen to be a name separator:
for hyphenated names, all components were included.
For metadata from PubMed and PMC where first (fore) and last names are coded separately, we applied the same cleaning steps.
We created <a href="https://git.dhimmel.com/pubmedpy/names.html">functions to simplify names</a> in the pubmedpy Python package to support standardized fore and last name processing.</p>
<h3 id="corresponding-author-extraction">Corresponding author extraction</h3>
<p>We assumed that, in the list of authors for a specific paper, corresponding authors (often research advisors) would be most likely to be invited for keynotes or to be honored as Fellows.
Therefore, we collected corresponding author names to assess the composition of the field, weighted by the number of corresponding authors per publication.</p>
<p>We evaluated two resources for extracting corresponding authors from papers: <a href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a> and <a href="https://www.ncbi.nlm.nih.gov/pmc/">PubMed Central</a> (PMC).
Both resources are provided by the US National Library of Medicine and index scholarly articles.
PubMed contains a record for every article published in journals it indexes (30 million records total circa 2020) and provides abstracts but not fulltext.
PMC, which provides fulltext access, does not contain every article from every journal (5.9 million records total circa 2020).
In general, open access journals will deposit their entire catalog to PMC (e.g., <em>BMC Bioinformatics</em> &amp; <em>PLOS Computational Biology</em>), while toll access journals (e.g., <em>Bioinformatics</em>) will only deposit articles when funders require it.
Since PMC requires publishers to submit fulltext articles in a structured XML format, the machine-readability and breadth of metadata in PMC is often superior to PubMed.</p>
<p>Of PMC’s 5.9 million fulltext articles, only 2.7 million are part of the “<a href="https://www.ncbi.nlm.nih.gov/pmc/tools/openftlist/">Open Access Subset</a>” which allows for downloading the structured fulltext as opposed to just viewing the article online.
However, authorship information does not require full text records.
We were able to download structured frontmatter (rather than fulltext) records from PMC’s <a href="https://www.ncbi.nlm.nih.gov/pmc/tools/oai/">OAI-PMH service</a>, so we were not limited to just the Open Access Subset.
For PubMed, we used the E-Utilities APIs.
For PubMed records, we were able to extract author first and last names and their order within a record.
For PMC, we were able to extract these fields as well as whether each author was a corresponding author.
To automate and generalize these tasks, we created the <a href="https://github.com/dhimmel/pubmedpy">pubmedpy</a> Python package.</p>
<p>We selected three journals to represent the field of bioinformatics and computational biology, including two ISCB Partner Journals (<em>PLOS Computational Biology</em> and <em>Bioinformatics</em>) and one field-specific journal that is not a partner (<em>BMC Bioinformatics</em>).
From PubMed, we compiled a catalog of 29,755 journal articles published from when each journal was established through 2019.
We were able to retrieve authorship information for all but 6 of these articles using PubMed or PubMed Central.</p>
<p>To determine corresponding authors for an article, we relied on PMC data if available (20,696 articles) and otherwise fell back to PubMed data (9,053 articles).
Almost all articles without PMC data were from <em>Bioinformatics</em> because it is a “selective deposit” rather than “full participation” journal in PMC.</p>
<p>We performed further analysis on PMC authors to learn more about corresponding author practices.
First, we developed and evaluated a method to infer a corresponding author when the coded corresponding status was not available.
For papers with multiple authors and at least one corresponding author, the first author was corresponding 43% of the time, whereas the last author was corresponding 62% of the time.
Therefore, we assumed the last author was corresponding when coded corresponding author status was not available (120 articles from PMC and all articles from PubMed).</p>
<p>Second, we investigated the number of corresponding authors for PMC articles.
81% of these articles had a single corresponding author.
1.7% had no corresponding authors.
Of these, many were editorials (e.g., <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1183510/">PMC1183510</a>, the announcement of <em>PLOS Computational Biology</em>).
A very small number of papers had over 10 corresponding authors.
Some of these instances were true outliers, like <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5001208/">PMC5001208</a> with 21 corresponding authors.
Others like <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3509495/">PMC3509495</a> were incorrect, due to upstream errors.
To not give undue influence to papers with multiple corresponding authors, subsequent analyses on corresponding authors are inversely weighted by the number of corresponding authors per paper.</p>
<h3 id="countries-of-affiliations">Countries of Affiliations</h3>
<p>Publications often provide affiliation lists for authors, which generally associate authors with research organizations and their corresponding physical addresses.
We implemented affiliation extraction in the pubmedpy Python package for both PubMed and PMC XML records.
These methods extract a sequence of textual affiliations for each author.
While ideally each affiliation record would refer to one and only one research organization, sometimes journals deposit multiple affiliations in a single structured affiliation.
For example, we extracted the following composite affiliation for all authors of <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4147893/">PMC4147893</a>:</p>
<blockquote>
<p>‘Multimodal Computing and Interaction’, Saarland University &amp; Department for Computational Biology and Applied Computing, Max Planck Institute for Informatics, Saarbrücken, 66123 Saarland, Germany, Ray and Stephanie Lane Center for Computational Biology, Carnegie Mellon University, Pittsburgh, 15206 PA, USA, Department of Mathematics and Computer Science, Freie Universität Berlin, 14195 Berlin, Germany, Université Pierre et Marie Curie, UMR7238, CNRS-UPMC, Paris, France and CNRS, UMR7238, Laboratory of Computational and Quantitative Biology, Paris, France.</p>
</blockquote>
<p>We designed a method for extracting countries from affiliations that accommodated multiple countries.
We relied on two Python utilities to extract countries from text: <a href="https://github.com/elyase/geotext">geotext</a> and <a href="https://geopy.readthedocs.io/en/stable/#nominatim"><code>geopy.geocoders.Nominatim</code></a>.
The first, geotext, used regular expressions to find mentions of places from the <a href="http://www.geonames.org/">GeoNames database</a>.
In the above text, geotext detected four mentions of places in Germany: Saarland, Saarbrücken, Saarland, Germany.
Anytime geotext identified 2 or more mentions of a country, we labeled the affiliation as including that country.</p>
<p><code>geopy.geocoders.Nominatim</code> converts names / addresses to geographic coordinates using the OpenStreetMap’s <a href="https://nominatim.org/">Nomatim</a> service.
We split textual affiliations by punctuation and found the first segment, in reverse order, that returned any Nomatim search results.
For the above affiliation, the search order was “France”, “Paris”, “Laboratory of Computational and Quantitative Biology”, etcetera.
Since searching “France” returns a match by Nomatim, the following queries would not be made.
When a match was found, we extracted the country containing the location.
This approach returns a single country for an affiliation when successful.
When labeling affiliations with countries, we only used these values when geotext did not return results or had ambiguity amongst countries without multiple matches.
For more details on this approach, consult the accompanying <a href="https://github.com/greenelab/iscb-diversity/blob/5213ba3451520af3967f74d8f58553dade0a826c/07.affiliations-to-countries.ipynb">notebook</a> and <a href="https://github.com/greenelab/iscb-diversity/blob/5213ba3451520af3967f74d8f58553dade0a826c/data/affiliations/geocode.jsonl">label dataset</a>.</p>
<p>Our ability to assign countries to authors was largely driven by the availability of affiliations.
In our catalog of corresponding authors, 95.7% of authors from PubMed Central records were assigned at least one country.
The country-assignment-rate for corresponding authors from PubMed records was only 32.2%.
This reflects the varying availability of affiliation metadata by journal.
Countries were assigned to all corresponding authors for 99.9% of articles in <em>BMC Bioinformatics</em> 98.7% in <em>PLOS Computational Biology</em>, and 49.6% in <em>Bioinformatics</em>.
Country assignment for corresponding authors in <em>Bioinformatics</em> were unavailable for over half of all articles annually from 1998–2013.
<em>Bioinformatics</em> affiliation metadata improved starting in 2014, and the country-assignment-rate has hovered around ~95% for the last few years (see <a href="https://github.com/greenelab/iscb-diversity/blob/a7fcc7c229672b56626f0dc68d04064079ff3a49/08.corresponding-authors.ipynb">notebook</a>).</p>
<p>For ISCB honorees, during the curation process, if an honoree was listed with their affiliation at the time, we recorded this affiliation for analysis.
For ISCB Fellows, we used the affiliation listed on the ISCB page.
Because we could not find affiliations for the 1997 and 1998 RECOMB keynote speakers’ listed for these years, they were left blank.
If an author or speaker had more than one affiliation, each was inversely weighted by the number of affiliations that individual had.</p>
<h3 id="estimation-of-gender">Estimation of Gender</h3>
<p>We predicted the gender of honorees and authors using the <a href="https://genderize.io" class="uri">https://genderize.io</a> API, which was trained on over 100 million name-gender pairings collected from the web and is one of the three widely-used gender inference services that provide gender classifications with over 98% accuracy <span class="citation" data-cites="ODb9u77d">[<a href="#ref-ODb9u77d" role="doc-biblioref">22</a>]</span>.
We used author and honoree first names to retrieve predictions from genderize.io.
The predictions represent the probability of an honoree or author being male or female.
We used the estimated probabilities and did not convert to a hard group assignment.
For example, a query to <a href="https://genderize.io" class="uri">https://genderize.io</a> on January 26, 2020 for “Casey” returns a probability of male of 0.74 and a probability of female of 0.26, which we would add for an author with this first name.
Because of the limitations of considering gender as a binary trait and inferring it from first names, we only consider predictions in aggregate and not as individual values for specific scientists.</p>
<p>Of 412 ISCB honorees, genderize.io fails to provide gender predictions for three names.
Of 34,005 corresponding authors, 45 were missing a fore name altogether in the raw paper metadata and 1,466 had fore names consisting of only initials.
Of the remaining authors, genderize.io failed to predict gender for 1,578 of these fore names.
We note that approximately 52% of these NA predictions are hyphenated names, which is likely because they are more unique and thus are more difficult to find predictions for.
79% of these names were predicted to be of Asian origin by last name (see the race/ethnicity prediction model below).
This bias of NA predictions toward non-English names has been previously observed <span class="citation" data-cites="cLM5pKQN">[<a href="#ref-cLM5pKQN" role="doc-biblioref">23</a>]</span> and may have a minor influence on the final estimate of gender compositions.</p>
<h3 id="estimation-of-name-origin-groups">Estimation of Name Origin Groups</h3>
<p>We developed a model to predict geographical origins of names.
The existing Python package ethnicolr <span class="citation" data-cites="1Fj9uUALi">[<a href="#ref-1Fj9uUALi" role="doc-biblioref">24</a>]</span> produces reasonable predictions, but its international representation in the data curated from Wikipedia in 2009 <span class="citation" data-cites="Wcublhdw">[<a href="#ref-Wcublhdw" role="doc-biblioref">25</a>]</span> is still limited.
For instance, 76% of the names in ethnicolr’s Wikipedia dataset are European in origin.</p>
<p>To address these limitations in ethnicolr, we built a similar classifier, a Long Short-term Memory (LSTM) neural network, to infer the region of origin from patterns in the sequences of letters in full names.
We applied this model on an updated, approximately 4.5 times larger training dataset called Wiki2019 (described below).
We tested multiple character sequence lengths and, based on this comparison, selected tri-characters for the primary results described in this work.
We trained our prediction model on 80% of the Wiki2019 dataset and evaluated its performance using the remaining 20%.
This model, which we term Wiki2019-LSTM, is available in the online file <a href="https://github.com/greenelab/wiki-nationality-estimate/blob/master/models/LSTM.h5"><code>LSTM.h5</code></a>.</p>
<p>To generate a training dataset for name origin prediction that reflects a modern naming landscape, we scraped the English Wikipedia’s category of <a href="https://en.wikipedia.org/wiki/Category:Living_people">Living People</a>.
This category, which contained approximately 930,000 pages at the time of processing in November 2019, is regularly curated and allowed us to avoid pages related to non-persons.
For each Wikipedia page, we used two strategies to find a full birth name and location context for that person.
First, we used information from the personal details sidebar; the information in this sidebar varied widely but often contained a full name and a place of birth.
Second, in the body of the text of most English-language biographical Wikipedia pages, the first sentence usually begins with, for example, “John Edward Smith (born 1 January 1970) is an American novelist known for …”
This structure comes from editor <a href="https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style/Biography#Context">guidance on biography articles</a> and is designed to capture:</p>
<blockquote>
<p>… the country of which the person is a citizen, national or permanent resident, or if the person is notable mainly for past events, the country where the person was a citizen, national or permanent resident when the person became notable.</p>
</blockquote>
<p>We used regular expressions to parse out the person’s name from this structure and checked that the expression after “is a” matched a list of nationalities.
We were able to define a name and nationality for 708,493 people by using the union of these strategies.
This process produced country labels that were more fine-grained than the broader patterns that we sought to examine among honorees and authors.
We initially grouped names by continent, but later decided to model our categorization after the hierarchical taxonomy used by <a href="http://www.name-prism.com/about">NamePrism</a> <span class="citation" data-cites="OqLU02w9">[<a href="#ref-OqLU02w9" role="doc-biblioref">26</a>]</span>.
The NamePrism taxonomy was derived from name-country pairs by producing an embedding of names by Twitter contact patterns and then grouping countries using the similarity of names from those countries.
The countries associated with each grouping are shown in Fig <a href="#fig:nameprism_countries">7</a>.
NamePrism excluded the US, Canada and Australia because these countries have been populated by a mix of immigrant groups <span class="citation" data-cites="OqLU02w9">[<a href="#ref-OqLU02w9" role="doc-biblioref">26</a>]</span>.</p>
<div id="fig:nameprism_countries" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/greenelab/iscb-diversity/3853a20773764cedcf6aa61f38f062b26bd7655d/figs/2020-01-31_groupings.png" style="width:90.0%" alt="" /><figcaption><span>Figure 7:</span> NamePrism groups countries by name similarity. We used this grouping but renamed the groups to focus on the linguistic patterns based on name etymology identified by NamePrism.</figcaption>
</figure>
</div>
<p>In an earlier version of this manuscript, we also used category names derived from NamePrism, but a reader <a href="https://github.com/greenelab/iscb-diversity-manuscript/issues/27">pointed out</a> the titles of the groupings were problematic;
therefore, in this version, we renamed these groupings to reflect that the NamePrism approach primarily identifies groups based on linguistic patterns from name etymology rather than religious or racial similarities.
We note that our mapping from nationality to name origins was not without error.
For example, a scientist of Israeli nationality may not bear a Hebrew name.
These mismatches were assessed via the heatmap of the model performance (Fig. <a href="#fig:wiki2019_lstm">2</a>C) and complemented by the affiliation analysis below.
An alternative approach is to assign arbitrary names to these groups such as via letter coding (e.g., A, B, C, etc.), but we did not choose this strategy because ten arbitrary letters for ten groups can greatly reduce the paper’s readibility.</p>
<h3 id="affiliation-analysis-1">Affiliation Analysis</h3>
<p>For each country, we computed the expected number of honorees by multiplying the proportion of authors whose affiliations were in that country with the total number of honorees.
We then performed an enrichment analysis to examine the difference in country affiliation proportions between ISCB honorees and field-specific corresponding authors.
We calculated each country’s enrichment by dividing the observed proportion of honorees by the expected proportion of honorees.
The 95% confidence interval of the log<sub>2</sub> enrichment was estimated using the Poisson model method <span class="citation" data-cites="wH0Hk1gE">[<a href="#ref-wH0Hk1gE" role="doc-biblioref">27</a>]</span>.</p>
<h3 id="estimation-of-race-and-ethnicity-within-the-us">Estimation of Race and Ethnicity within the US</h3>
<p>We predicted the race and ethnicity of US-affiliated honorees and authors using the R package wru.
wru implements methods described in Imai and Khanna <span class="citation" data-cites="VGmZCDFw">[<a href="#ref-VGmZCDFw" role="doc-biblioref">28</a>]</span> to predict race and ethnicity using surname and location information.
The underlying data used for prediction are derived from the US Census, in which an individual’s race and ethnicity are based on their self-identification with one or more groups.
Specifically, the race categories include White, Black or African American, American Indian or Alaska Native, Asian, Native Hawaiian or Other Pacific Islander, Other race, and Two or more races <span class="citation" data-cites="Hhiridja">[<a href="#ref-Hhiridja" role="doc-biblioref">29</a>]</span>, and ethnicity categories include Hispanic/Latino or Not Hispanic/Latino <span class="citation" data-cites="KcIUCUC7">[<a href="#ref-KcIUCUC7" role="doc-biblioref">30</a>]</span>.
wru uses similar race/ethnicity categories but groups American Indian or Alaska Native and Native Hawaiian or Other Pacific Islander to form the Other category.</p>
<p>We used only the surname of author or honoree to make predictions via the <em>predict_race()</em> function from wru.
However, in the case of names that were not observed in the census, the function outputs the average demographic distribution from the census, which may produce misleading results.
To avoid this suboptimal imputation, we modified the function to return a status denoting that results were inconclusive (NA) instead.
This prediction represents the probability of an honoree or author selecting a certain race or ethnicity on a census form if they lived within the US.</p>
<p>Of 239 US-affiliated ISCB honoree entries, wru fails to provide race/ethnicity predictions for 45 names.
Of 11,545 US-affiliated corresponding authors, 2,352 had a last name for which wru did not provide predictions.
One limitation of wru and other methods that infer race, ethnicity, or nationality from last names is the potentially inaccurate prediction for scientists who changed their last name during marriage, a practice more common among women than men.</p>
<h2 id="data-and-resource-availability">Data and Resource Availability</h2>
<p>This manuscript was written <a href="https://github.com/greenelab/iscb-diversity-manuscript">openly on GitHub</a> using Manubot <span class="citation" data-cites="YuJbg3zO">[<a href="#ref-YuJbg3zO" role="doc-biblioref">31</a>]</span>.
The Manubot HTML version is available under a Creative Commons Attribution (CC BY 4.0) License at <a href="https://greenelab.github.io/iscb-diversity-manuscript/" class="uri">https://greenelab.github.io/iscb-diversity-manuscript/</a>.
Our analysis of authors and ISCB-associated honorees is available under CC BY 4.0 at <a href="https://github.com/greenelab/iscb-diversity" class="uri">https://github.com/greenelab/iscb-diversity</a>, with source code also distributed under a BSD 3-Clause License.
Rendered Python and R notebooks from this repository are browsable at <a href="https://greenelab.github.io/iscb-diversity/" class="uri">https://greenelab.github.io/iscb-diversity/</a>.
Our analysis of PubMed, PubMed Central, and author names relies on the Python pubmedpy package, developed as part of this project and available under a Blue Oak Model License 1.0 at <a href="https://github.com/dhimmel/pubmedpy" class="uri">https://github.com/dhimmel/pubmedpy</a> and on <a href="https://pypi.org/project/pubmedpy/">PyPI</a>.
Our Wikipedia name dataset is dedicated to the public domain under CC0 License at <a href="https://github.com/greenelab/wiki-nationality-estimate" class="uri">https://github.com/greenelab/wiki-nationality-estimate</a>, with source code to construct the dataset available under a BSD 3-Clause License.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references hanging-indent" role="doc-bibliography">
<div id="ref-hP9R3Quu">
<p>1. <strong>Topic choice contributes to the lower rate of NIH awards to African-American/black scientists</strong><br />
Travis A. Hoppe, Aviva Litovitz, Kristine A. Willis, Rebecca A. Meseroll, Matthew J. Perkins, B. Ian Hutchins, Alison F. Davis, Michael S. Lauer, Hannah A. Valantine, James M. Anderson, George M. Santangelo<br />
<em>Science Advances</em> (2019-10-09) <a href="https://doi.org/gghp8t">https://doi.org/gghp8t</a><br />
DOI: <a href="https://doi.org/10.1126/sciadv.aaw7238">10.1126/sciadv.aaw7238</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31633016">31633016</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6785250">PMC6785250</a></p>
</div>
<div id="ref-JHzGjxks">
<p>2. <strong>Ten Simple Rules to Achieve Conference Speaker Gender Balance</strong><br />
Jennifer L. Martin<br />
<em>PLoS Computational Biology</em> (2014-11-20) <a href="https://doi.org/gf853n">https://doi.org/gf853n</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1003903">10.1371/journal.pcbi.1003903</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25411977">25411977</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4238945">PMC4238945</a></p>
</div>
<div id="ref-w8Lq7zPE">
<p>3. <strong>How scientists are fighting against gender bias in conference speaker lineups</strong><br />
Katie Langin<br />
<em>Science</em> (2019-02-11) <a href="https://doi.org/gghp8v">https://doi.org/gghp8v</a><br />
DOI: <a href="https://doi.org/10.1126/science.caredit.aaw9742">10.1126/science.caredit.aaw9742</a></p>
</div>
<div id="ref-1G16T65Sv">
<p>4. <strong>Speaking out about gender imbalance in invited speakers improves diversity</strong><br />
Robyn S Klein, Rhonda Voskuhl, Benjamin M Segal, Bonnie N Dittel, Thomas E Lane, John R Bethea, Monica J Carson, Carol Colton, Susanna Rosi, Aileen Anderson, … Anne H Cross<br />
<em>Nature Immunology</em> (2017-05-01) <a href="https://doi.org/gghp8s">https://doi.org/gghp8s</a><br />
DOI: <a href="https://doi.org/10.1038/ni.3707">10.1038/ni.3707</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28418385">28418385</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5775963">PMC5775963</a></p>
</div>
<div id="ref-1Hmlj5vSu">
<p>5. <strong>Addressing the underrepresentation of women in mathematics conferences</strong><br />
Greg Martin<br />
<em>arXiv</em> (2015-02-24) <a href="https://arxiv.org/abs/1502.06326">https://arxiv.org/abs/1502.06326</a></p>
</div>
<div id="ref-UxTZTP0G">
<p>6. <strong>The Presence of Female Conveners Correlates with a Higher Proportion of Female Speakers at Scientific Symposia</strong><br />
Arturo Casadevall, Jo Handelsman<br />
<em>mBio</em> (2014-01-07) <a href="https://doi.org/qsh">https://doi.org/qsh</a><br />
DOI: <a href="https://doi.org/10.1128/mbio.00846-13">10.1128/mbio.00846-13</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24399856">24399856</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3884059">PMC3884059</a></p>
</div>
<div id="ref-6mHXFr1Z">
<p>7. <strong>Trends in the Proportion of Female Speakers at Medical Conferences in the United States and in Canada, 2007 to 2017</strong><br />
Shannon M. Ruzycki, Sarah Fletcher, Madalene Earp, Aleem Bharwani, Kirstie C. Lithgow<br />
<em>JAMA Network Open</em> (2019-04-12) <a href="https://doi.org/gghp8r">https://doi.org/gghp8r</a><br />
DOI: <a href="https://doi.org/10.1001/jamanetworkopen.2019.2103">10.1001/jamanetworkopen.2019.2103</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30977853">30977853</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6481599">PMC6481599</a></p>
</div>
<div id="ref-19KEajiJN">
<p>8. <strong>Are gender gaps due to evaluations of the applicant or the science? A natural experiment at a national funding agency</strong><br />
Holly O Witteman, Michael Hendricks, Sharon Straus, Cara Tannenbaum<br />
<em>The Lancet</em> (2019-02) <a href="https://doi.org/djc5">https://doi.org/djc5</a><br />
DOI: <a href="https://doi.org/10.1016/s0140-6736(18)32611-4">10.1016/s0140-6736(18)32611-4</a></p>
</div>
<div id="ref-njNa3CYa">
<p>9. <strong>Race, Ethnicity, and NIH Research Awards</strong><br />
D. K. Ginther, W. T. Schaffer, J. Schnell, B. Masimore, F. Liu, L. L. Haak, R. Kington<br />
<em>Science</em> (2011-08-18) <a href="https://doi.org/csf8j8">https://doi.org/csf8j8</a><br />
DOI: <a href="https://doi.org/10.1126/science.1196783">10.1126/science.1196783</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21852498">21852498</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3412416">PMC3412416</a></p>
</div>
<div id="ref-167tJVlJu">
<p>10. <strong>Gender, Race/Ethnicity, and National Institutes of Health R01 Research Awards</strong><br />
Donna K. Ginther, Shulamit Kahn, Walter T. Schaffer<br />
<em>Academic Medicine</em> (2016-08) <a href="https://doi.org/f8zzw4">https://doi.org/f8zzw4</a><br />
DOI: <a href="https://doi.org/10.1097/acm.0000000000001278">10.1097/acm.0000000000001278</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27306969">27306969</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4965301">PMC4965301</a></p>
</div>
<div id="ref-14PjsfIPd">
<p>11. <strong>Persistent Underrepresentation of Women’s Science in High Profile Journals</strong><br />
Yiqin Alicia Shen, Jason M. Webster, Yuichi Shoda, Ione Fine<br />
<em>bioRxiv</em> (2018-03-08) <a href="https://doi.org/cmh5">https://doi.org/cmh5</a><br />
DOI: <a href="https://doi.org/10.1101/275362">10.1101/275362</a></p>
</div>
<div id="ref-17aOPYsbT">
<p>12. <strong>The gender gap in science: How long until women are equally represented?</strong><br />
Luke Holman, Devi Stuart-Fox, Cindy E. Hauser<br />
<em>PLOS Biology</em> (2018-04-19) <a href="https://doi.org/gdb9db">https://doi.org/gdb9db</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pbio.2004956">10.1371/journal.pbio.2004956</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29672508">29672508</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5908072">PMC5908072</a></p>
</div>
<div id="ref-9TNbcewU">
<p>13. <strong>The extent and drivers of gender imbalance in neuroscience reference lists</strong><br />
Jordan D. Dworkin, Kristin A. Linn, Erin G. Teich, Perry Zurn, Russell T. Shinohara, Danielle S. Bassett<br />
<em>arXiv</em> (2020-04-10) <a href="https://arxiv.org/abs/2001.01002">https://arxiv.org/abs/2001.01002</a></p>
</div>
<div id="ref-5HWeup6K">
<p>14. <strong>Gender differences in peer review outcomes and manuscript impact at six journals of ecology and evolution</strong><br />
Charles W. Fox, C. E. Timothy Paine<br />
<em>Ecology and Evolution</em> (2019-03-04) <a href="https://doi.org/gfwjjb">https://doi.org/gfwjjb</a><br />
DOI: <a href="https://doi.org/10.1002/ece3.4993">10.1002/ece3.4993</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30962913">30962913</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6434606">PMC6434606</a></p>
</div>
<div id="ref-1EYk4f8Mu">
<p>15. <strong>Journals invite too few women to referee</strong><br />
Jory Lerback, Brooks Hanson<br />
<em>Nature</em> (2017-01-26) <a href="https://doi.org/gf4jjz">https://doi.org/gf4jjz</a><br />
DOI: <a href="https://doi.org/10.1038/541455a">10.1038/541455a</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28128272">28128272</a></p>
</div>
<div id="ref-1FX4fLEIy">
<p>16. <strong>Role modeling is a viable retention strategy for undergraduate women in the geosciences</strong><br />
Paul R. Hernandez, Brittany Bloodhart, Amanda S. Adams, Rebecca T. Barnes, Melissa Burt, Sandra M. Clinton, Wenyi Du, Elaine Godfrey, Heather Henderson, Ilana B. Pollack, Emily V. Fischer<br />
<em>Geosphere</em> (2018-10-31) <a href="https://doi.org/gghp9d">https://doi.org/gghp9d</a><br />
DOI: <a href="https://doi.org/10.1130/ges01659.1">10.1130/ges01659.1</a></p>
</div>
<div id="ref-zdS2Wmy2">
<p>17. <strong>Why we’re editing women scientists onto Wikipedia</strong><br />
Jess Wade, Maryam Zaringhalam<br />
<em>Nature</em> (2018-08-14) <a href="https://doi.org/gdz52z">https://doi.org/gdz52z</a><br />
DOI: <a href="https://doi.org/10.1038/d41586-018-05947-8">10.1038/d41586-018-05947-8</a></p>
</div>
<div id="ref-yaljvCUU">
<p>18. <strong>Why we’re creating Wikipedia profiles for BAME scientists</strong><br />
Nicola O’Reilly<br />
<em>Nature</em> (2019-03-07) <a href="https://doi.org/gfwhcr">https://doi.org/gfwhcr</a><br />
DOI: <a href="https://doi.org/10.1038/d41586-019-00812-8">10.1038/d41586-019-00812-8</a></p>
</div>
<div id="ref-1Bji9UOxF">
<p>19. <strong>Research in Computational Molecular Biology</strong><br />
Lecture Notes in Computer Science<br />
(2016) <a href="https://doi.org/gghp87">https://doi.org/gghp87</a><br />
DOI: <a href="https://doi.org/10.1007/978-3-319-31957-5">10.1007/978-3-319-31957-5</a></p>
</div>
<div id="ref-DBJsvxt7">
<p>20. <strong>Research in Computational Molecular Biology</strong><br />
Lecture Notes in Computer Science<br />
(2011) <a href="https://doi.org/dvwjgg">https://doi.org/dvwjgg</a><br />
DOI: <a href="https://doi.org/10.1007/978-3-642-20036-6">10.1007/978-3-642-20036-6</a></p>
</div>
<div id="ref-aziMb8CD">
<p>21. <strong>Research in Computational Molecular Biology</strong><br />
Lecture Notes in Computer Science<br />
(2015) <a href="https://doi.org/gghwjh">https://doi.org/gghwjh</a><br />
DOI: <a href="https://doi.org/10.1007/978-3-319-16706-0">10.1007/978-3-319-16706-0</a></p>
</div>
<div id="ref-ODb9u77d">
<p>22. <strong>Comparison and benchmark of name-to-gender inference services</strong><br />
Lucía Santamaría, Helena Mihaljević<br />
<em>PeerJ Computer Science</em> (2018-07-16) <a href="https://doi.org/ggpbn6">https://doi.org/ggpbn6</a><br />
DOI: <a href="https://doi.org/10.7717/peerj-cs.156">10.7717/peerj-cs.156</a></p>
</div>
<div id="ref-cLM5pKQN">
<p>23. <strong>Gender Prediction Methods Based on First Names with genderizeR</strong><br />
Kamil Wais<br />
<em>The R Journal</em> (2016) <a href="https://doi.org/gf4zqx">https://doi.org/gf4zqx</a><br />
DOI: <a href="https://doi.org/10.32614/rj-2016-002">10.32614/rj-2016-002</a></p>
</div>
<div id="ref-1Fj9uUALi">
<p>24. <strong>Predicting Race and Ethnicity From the Sequence of Characters in a Name</strong><br />
Gaurav Sood, Suriyan Laohaprapanon<br />
<em>arXiv</em> (2018-07-31) <a href="https://arxiv.org/abs/1805.02109">https://arxiv.org/abs/1805.02109</a></p>
</div>
<div id="ref-Wcublhdw">
<p>25. <strong>Name-ethnicity classification from open sources</strong><br />
Anurag Ambekar, Charles Ward, Jahangir Mohammed, Swapna Male, Steven Skiena<br />
<em>Association for Computing Machinery (ACM)</em> (2009) <a href="https://doi.org/fs3pr8">https://doi.org/fs3pr8</a><br />
DOI: <a href="https://doi.org/10.1145/1557019.1557032">10.1145/1557019.1557032</a></p>
</div>
<div id="ref-OqLU02w9">
<p>26. <strong>Nationality Classification Using Name Embeddings</strong><br />
Junting Ye, Shuchu Han, Yifan Hu, Baris Coskun, Meizhu Liu, Hong Qin, Steven Skiena<br />
<em>Association for Computing Machinery (ACM)</em> (2017) <a href="https://doi.org/ggjc78">https://doi.org/ggjc78</a><br />
DOI: <a href="https://doi.org/10.1145/3132847.3133008">10.1145/3132847.3133008</a></p>
</div>
<div id="ref-wH0Hk1gE">
<p>27. <strong>Statistics in epidemiology: methods, techniques, and applications</strong><br />
Hardeo Sahai, Anwer Khurshid<br />
<em>CRC Press</em> (1996)<br />
ISBN: <a href="https://worldcat.org/isbn/9780849394447">9780849394447</a></p>
</div>
<div id="ref-VGmZCDFw">
<p>28. <strong>Improving Ecological Inference by Predicting Individual Ethnicity from Voter Registration Records</strong><br />
Kosuke Imai, Kabir Khanna<br />
<em>Political Analysis</em> (2017-01-04) <a href="https://doi.org/f8ntmv">https://doi.org/f8ntmv</a><br />
DOI: <a href="https://doi.org/10.1093/pan/mpw001">10.1093/pan/mpw001</a></p>
</div>
<div id="ref-Hhiridja">
<p>29. <a href="https://data.census.gov/cedsci/deeplinks?url=https://factfinder.census.gov/help/en/race.htm">https://data.census.gov/cedsci/deeplinks?url=https://factfinder.census.gov/help/en/race.htm</a></p>
</div>
<div id="ref-KcIUCUC7">
<p>30. <strong>Census Bureau News</strong>(2001-04-05) <a href="https://web.archive.org/web/20010405061504/http://www.census.gov/Press-Release/www/2001/raceqandas.html">https://web.archive.org/web/20010405061504/http://www.census.gov/Press-Release/www/2001/raceqandas.html</a></p>
</div>
<div id="ref-YuJbg3zO">
<p>31. <strong>Open collaborative writing with Manubot</strong><br />
Daniel S. Himmelstein, Vincent Rubinetti, David R. Slochower, Dongbo Hu, Venkat S. Malladi, Casey S. Greene, Anthony Gitter<br />
<em>PLOS Computational Biology</em> (2019-06-24) <a href="https://doi.org/c7np">https://doi.org/c7np</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1007128">10.1371/journal.pcbi.1007128</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31233491">31233491</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6611653">PMC6611653</a></p>
</div>
</div>
<!-- default theme -->

<style>
    /* import google fonts */
    @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
    @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

    /* -------------------------------------------------- */
    /* global */
    /* -------------------------------------------------- */

    /* all elements */
    * {
        /* force sans-serif font unless specified otherwise */
        font-family: "Open Sans", "Helvetica", sans-serif;

        /* prevent text inflation on some mobile browsers */
        -webkit-text-size-adjust: none !important;
        -moz-text-size-adjust: none !important;
        -o-text-size-adjust: none !important;
        text-size-adjust: none !important;
    }

    @media only screen {
        /* "page" element */
        body {
            position: relative;
            box-sizing: border-box;
            font-size: 12pt;
            line-height: 1.5;
            max-width: 8.5in;
            margin: 20px auto;
            padding: 40px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* "page" element */
        body {
            padding: 20px;
            margin: 0;
            border-radius: 0;
            border: none;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
            background: none;
        }
    }

    /* -------------------------------------------------- */
    /* headings */
    /* -------------------------------------------------- */

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin: 20px 0;
        padding: 0;
        font-weight: bold;
    }

    /* biggest heading */
    h1 {
        margin: 40px 0;
        text-align: center;
    }

    /* second biggest heading */
    h2 {
        margin-top: 30px;
        padding-bottom: 5px;
        border-bottom: solid 1px #bdbdbd;
    }

    /* -------------------------------------------------- */
    /* manuscript header */
    /* -------------------------------------------------- */

    /* manuscript title */
    header > h1 {
        margin: 0;
    }

    /* manuscript title caption text (ie "automatically generated on") */
    header + p {
        text-align: center;
        margin-top: 10px;
    }

    /* -------------------------------------------------- */
    /* text elements */
    /* -------------------------------------------------- */

    /* links */
    a {
        color: #2196f3;
        overflow-wrap: break-word;
    }

    /* normal links (not empty, not button link, not syntax highlighting link) */
    a:not(:empty):not(.button):not(.sourceLine) {
        padding-left: 1px;
        padding-right: 1px;
    }

    /* superscripts and subscripts */
    sub,
    sup {
        /* prevent from affecting line height */
        line-height: 0;
    }

    /* unordered and ordered lists*/
    ul,
    ol {
        padding-left: 20px;
    }

    /* class for styling text semibold */
    .semibold {
        font-weight: 600;
    }

    /* class for styling elements horizontally left aligned */
    .left {
        display: block;
        text-align: left;
        margin-left: auto;
        margin-right: 0;
        justify-content: left;
    }

    /* class for styling elements horizontally centered */
    .center {
        display: block;
        text-align: center;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
    }

    /* class for styling elements horizontally right aligned */
    .right {
        display: block;
        text-align: right;
        margin-left: 0;
        margin-right: auto;
        justify-content: right;
    }

    /* -------------------------------------------------- */
    /* section elements */
    /* -------------------------------------------------- */

    /* horizontal divider line */
    hr {
        border: none;
        height: 1px;
        background: #bdbdbd;
    }

    /* paragraphs, horizontal dividers, figures, tables, code */
    p,
    hr,
    figure,
    table,
    pre {
        /* treat all as "paragraphs", with consistent vertical margins */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* compact author list */
    /* -------------------------------------------------- */
    #authors + ul p {
        margin-top: 2px;
        margin-bottom: 6px;
    }

    /* -------------------------------------------------- */
    /* compact references */
    /* -------------------------------------------------- */
    #refs {
        font-size: 0.85em;
    }
    #refs p {
        margin-top: 8px;
        margin-bottom: 8px;
    }

    /* -------------------------------------------------- */
    /* figures */
    /* -------------------------------------------------- */

    /* figure */
    figure {
        max-width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure caption */
    figcaption {
        padding: 0;
        padding-top: 10px;
    }

    /* figure image element */
    figure img {
        max-width: 100%;
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure auto-number */
    img + figcaption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* tables */
    /* -------------------------------------------------- */

    /* table */
    table {
        border-collapse: collapse;
        border-spacing: 0;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* table cells */
    th,
    td {
        border: solid 1px #bdbdbd;
        padding: 10px;
        /* squash table if too wide for page by forcing line breaks */
        overflow-wrap: break-word;
        word-break: break-word;
    }

    /* header row and even rows */
    th,
    tr:nth-child(2n) {
        background-color: #fafafa;
    }

    /* odd rows */
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    }

    /* table caption */
    caption {
        text-align: left;
        padding: 0;
        padding-bottom: 10px;
    }

    /* table auto-number */
    table > caption > span:first-of-type,
    div.table_wrapper > table > caption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* code */
    /* -------------------------------------------------- */

    /* multi-line code block */
    pre {
        padding: 10px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
        break-inside: avoid;
        text-align: left;
    }

    /* inline code, ie code within normal text */
    :not(pre) > code {
        padding: 0 4px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
    }

    /* code text */
    /* apply all children, to reach syntax highlighting sub-elements */
    code,
    code * {
        /* force monospace font */
        font-family: "Source Code Pro", "Courier New", monospace;
    }

    /* -------------------------------------------------- */
    /* quotes */
    /* -------------------------------------------------- */

    /* quoted text */
    blockquote {
        margin: 0;
        padding: 0;
        border-left: 4px solid #bdbdbd;
        padding-left: 16px;
        break-inside: avoid;
    }

    /* -------------------------------------------------- */
    /* banners */
    /* -------------------------------------------------- */

    /* info banners */
    .banner {
        box-sizing: border-box;
        display: block;
        position: relative;
        width: 100%;
        margin-top: 20px;
        margin-bottom: 20px;
        padding: 20px;
        text-align: center;
    }

    /* paragraph in banner */
    .banner > p {
        margin: 0;
    }

    /* -------------------------------------------------- */
    /* highlight colors */
    /* -------------------------------------------------- */

    .white {
        background: #ffffff;
    }
    .lightgrey {
        background: #eeeeee;
    }
    .grey {
        background: #757575;
    }
    .darkgrey {
        background: #424242;
    }
    .black {
        background: #000000;
    }
    .lightred {
        background: #ffcdd2;
    }
    .lightyellow {
        background: #ffecb3;
    }
    .lightgreen {
        background: #dcedc8;
    }
    .lightblue {
        background: #e3f2fd;
    }
    .lightpurple {
        background: #f3e5f5;
    }
    .red {
        background: #f44336;
    }
    .orange {
        background: #ff9800;
    }
    .yellow {
        background: #ffeb3b;
    }
    .green {
        background: #4caf50;
    }
    .blue {
        background: #2196f3;
    }
    .purple {
        background: #9c27b0;
    }
    .white,
    .lightgrey,
    .lightred,
    .lightyellow,
    .lightgreen,
    .lightblue,
    .lightpurple,
    .orange,
    .yellow,
    .white a,
    .lightgrey a,
    .lightred a,
    .lightyellow a,
    .lightgreen a,
    .lightblue a,
    .lightpurple a,
    .orange a,
    .yellow a {
        color: #000000;
    }
    .grey,
    .darkgrey,
    .black,
    .red,
    .green,
    .blue,
    .purple,
    .grey a,
    .darkgrey a,
    .black a,
    .red a,
    .green a,
    .blue a,
    .purple a {
        color: #ffffff;
    }

    /* -------------------------------------------------- */
    /* buttons */
    /* -------------------------------------------------- */

    /* class for styling links like buttons */
    .button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        margin: 5px;
        padding: 10px 20px;
        font-size: 0.75em;
        font-weight: 600;
        text-transform: uppercase;
        text-decoration: none;
        letter-spacing: 1px;
        background: none;
        color: #2196f3;
        border: solid 1px #bdbdbd;
        border-radius: 5px;
    }

    /* buttons when hovered */
    .button:hover:not([disabled]),
    .icon_button:hover:not([disabled]) {
        cursor: pointer;
        background: #f5f5f5;
    }

    /* buttons when disabled */
    .button[disabled],
    .icon_button[disabled] {
        opacity: 0.35;
        pointer-events: none;
    }

    /* class for styling buttons containg only single icon */
    .icon_button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        margin: 0;
        padding: 0;
        background: none;
        border-radius: 5px;
        border: none;
        width: 20px;
        height: 20px;
        min-width: 20px;
        min-height: 20px;
    }

    /* icon button inner svg image */
    .icon_button > svg {
        height: 16px;
    }

    /* -------------------------------------------------- */
    /* icons */
    /* -------------------------------------------------- */

    /* class for styling icons inline with text */
    .inline_icon {
        height: 1em;
        position: relative;
        top: 0.125em;
    }

    /* -------------------------------------------------- */
    /* print control */
    /* -------------------------------------------------- */

    @media print {
        @page {
            /* suggested printing margin */
            margin: 0.5in;
        }

        /* document and "page" elements */
        html, body {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }

        /* "page" element */
        body {
            font-size: 10.7pt !important;
            line-height: 1.35;
        }

        /* all headings */
        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin: 8px 0;
        }

        /* heading 1 */
        h1 {
            font-size: 1.6em;
        }

        /* heading 2 */
        h2 {
            font-size: 1.2em;
            margin-top: 0;
        }

        /* heading 3 */
        h3 {
            font-size: 1.10em;
        }

        /* figures and tables */
        figure, table {
            font-size: 0.85em;
        }

        /* table cells */
        th,
        td {
            padding: 5px;
        }

        /* shrink font awesome icons */
        i.fas,
        i.fab,
        i.far,
        i.fal {
            transform: scale(0.85);
        }

        /* decrease banner margins */
        .banner {
            margin-top: 15px;
            margin-bottom: 15px;
            padding: 15px;
        }

        /* class for centering an element vertically on its own page */
        .page_center {
            margin: auto;
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            vertical-align: middle;
            break-before: page;
            break-after: page;
        }

        /* always insert a page break before the element */
        .page_break_before {
            break-before: page;
        }

        /* always insert a page break after the element */
        .page_break_after {
            break-after: page;
        }

        /* avoid page break before the element */
        .page_break_before_avoid {
            break-before: avoid;
        }

        /* avoid page break after the element */
        .page_break_after_avoid {
            break-after: avoid;
        }

        /* avoid page break inside the element */
        .page_break_inside_avoid {
            break-inside: avoid;
        }
    }

    /* -------------------------------------------------- */
    /* override pandoc css quirks */
    /* -------------------------------------------------- */

    .sourceCode {
        /* prevent unsightly overflow in wide code blocks */
        overflow: auto !important;
    }

    div.sourceCode {
        /* prevent background fill on top-most code block  container */
        background: none !important;
    }

    .sourceCode * {
        /* force consistent line spacing */
        line-height: 1.5 !important;
    }

    div.sourceCode {
        /* style code block margins same as <pre> element */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* tablenos */
    /* -------------------------------------------------- */

    /* tablenos wrapper */
    .tablenos {
        /* show scrollbar on tables if necessary to prevent overflow */
        width: 100%;
        margin: 20px 0;
    }

    .tablenos > table {
        /* move margins from table to table_wrapper to allow margin collapsing */
        margin: 0;
    }

    @media only screen {
        /* tablenos wrapper */
        .tablenos {
            /* show scrollbar on tables if necessary to prevent overflow */
            overflow-x: auto !important;
        }

        .tablenos th,
        .tablenos td {
            overflow-wrap: unset !important;
            word-break: unset !important;
        }

        /* table in wrapper */
        .tablenos table,
        .tablenos table * {
            /* don't break table words */
            overflow-wrap: normal !important;
        }
    }

    /* -------------------------------------------------- */
    /* mathjax */
    /* -------------------------------------------------- */

    /* mathjax containers */
    .math.display > span:not(.MathJax_Preview) {
        /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
        display: flex !important;
        overflow-x: auto !important;
        overflow-y: hidden !important;
        justify-content: center;
        align-items: center;
        margin: 0 !important;
    }

    /* right click menu */
    .MathJax_Menu {
        border-radius: 5px !important;
        border: solid 1px #bdbdbd !important;
        box-shadow: none !important;
    }

    /* equation auto-number */
    span[id^="eq:"] > span.math.display + span {
        font-weight: 600;
    }

    /* equation */
    span[id^="eq:"] > span.math.display > span {
        /* nudge to make room for equation auto-number and anchor */
        margin-right: 60px !important;
    }

    /* -------------------------------------------------- */
    /* anchors plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anchor button */
        .anchor {
            opacity: 0;
            margin-left: 5px;
        }

        /* anchor buttons within <h2>'s */
        h2 .anchor {
            margin-left: 10px;
        }

        /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
        *:hover > .anchor,
        .anchor:hover,
        .anchor:focus {
            opacity: 1;
        }

        /* anchor button when hovered */
        .anchor:hover {
            cursor: pointer;
        }
    }

    /* always show anchor button on devices with no mouse/hover ability */
    @media (hover: none) {
        .anchor {
            opacity: 1;
        }
    }

    /* always hide anchor button on print */
    @media only print {
        .anchor {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* accordion plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* accordion arrow button */
        .accordion_arrow {
            margin-right: 10px;
        }

        /* arrow icon when <h2> data-collapsed attribute true */
        h2[data-collapsed="true"] > .accordion_arrow > svg {
            transform: rotate(-90deg);
        }

        /* all elements (except <h2>'s) when data-collapsed attribute true */
        *:not(h2)[data-collapsed="true"] {
            display: none;
        }

        /* accordion arrow button when hovered and <h2>'s when hovered */
        .accordion_arrow:hover,
        h2[data-collapsed="true"]:hover,
        h2[data-collapsed="false"]:hover {
            cursor: pointer;
        }
    }

    /* always hide accordion arrow button on print */
    @media only print {
        .accordion_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* tooltips plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* tooltip container */
        #tooltip {
            position: absolute;
            width: 50%;
            min-width: 240px;
            max-width: 75%;
            z-index: 1;
        }

        /* tooltip content */
        #tooltip_content {
            margin-bottom: 5px;
            padding: 20px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
            overflow-wrap: break-word;
        }

        /* tooltip copy of paragraphs and figures */
        #tooltip_content > p,
        #tooltip_content > figure {
            margin: 0;
            max-height: 320px;
            overflow-y: auto;
        }

        /* tooltip copy of <img> */
        #tooltip_content > figure > img {
            max-height: 260px;
        }

        /* navigation bar */
        #tooltip_nav_bar {
            margin-top: 10px;
            text-align: center;
        }

        /* navigation bar previous/next buton */
        #tooltip_nav_bar > .icon_button {
            position: relative;
            top: 3px;
        }

        /* navigation bar previous button */
        #tooltip_nav_bar > .icon_button:first-of-type {
            margin-right: 5px;
        }

        /* navigation bar next button */
        #tooltip_nav_bar > .icon_button:last-of-type {
            margin-left: 5px;
        }
    }

    /* always hide tooltip on print */
    @media only print {
        #tooltip {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* jump to first plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* jump button */
        .jump_arrow {
            position: relative;
            top: 0.125em;
            margin-right: 5px;
        }
    }

    /* always hide jump button on print */
    @media only print {
        .jump_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* link highlight plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anything with data-highlighted attribute true */
        [data-highlighted="true"] {
            background: #ffeb3b;
        }

        /* anything with data-selected attribute true */
        [data-selected="true"] {
            background: #ff8a65 !important;
        }

        /* animation definition for glow */
        @keyframes highlight_glow {
            0% {
                background: none;
            }
            10% {
                background: #bbdefb;
            }
            100% {
                background: none;
            }
        }

        /* anything with data-glow attribute true */
        [data-glow="true"] {
            animation: highlight_glow 2s;
        }
    }

    /* -------------------------------------------------- */
    /* table of contents plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* toc panel */
        #toc_panel {
            box-sizing: border-box;
            position: fixed;
            top: 0;
            left: 0;
            background: #ffffff;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            z-index: 2;
        }

        /* toc panel when closed */
        #toc_panel[data-open="false"] {
            min-width: 60px;
            width: 60px;
            height: 60px;
            border-right: solid 1px #bdbdbd;
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc panel when open */
        #toc_panel[data-open="true"] {
            min-width: 260px;
            max-width: 480px;
            /* keep panel edge consistent distance away from "page" edge */
            width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
            bottom: 0;
            border-right: solid 1px #bdbdbd;
        }

        /* toc panel header */
        #toc_header {
            box-sizing: border-box;
            display: flex;
            flex-direction: row;
            align-items: center;
            height: 60px;
            margin: 0;
            padding: 20px;
        }

        /* toc panel header when hovered */
        #toc_header:hover {
            cursor: pointer;
        }

        /* toc panel header when panel open */
        #toc_panel[data-open="true"] > #toc_header {
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc open/close header button */
        #toc_button {
            margin-right: 20px;
        }

        /* hide toc list and header text when closed */
        #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
        #toc_panel[data-open="false"] > #toc_list {
            display: none;
        }

        /* toc list of entries */
        #toc_list {
            box-sizing: border-box;
            width: 100%;
            padding: 20px;
            position: absolute;
            top: calc(60px + 1px);
            bottom: 0;
            overflow: auto;
        }

        /* toc entry, link to section in document */
        .toc_link {
            display: block;
            padding: 5px;
            position: relative;
            font-weight: 600;
            text-decoration: none;
        }

        /* toc entry when hovered or when "viewed" */
        .toc_link:hover,
        .toc_link[data-viewing="true"] {
            background: #f5f5f5;
        }

        /* toc entry, level 1 indentation */
        .toc_link[data-level="1"] {
            margin-left: 0;
        }

        /* toc entry, level 2 indentation */
        .toc_link[data-level="2"] {
            margin-left: 20px;
        }

        /* toc entry, level 3 indentation */
        .toc_link[data-level="3"] {
            margin-left: 40px;
        }

        /* toc entry, level 4 indentation */
        .toc_link[data-level="4"] {
            margin-left: 60px;
        }

        /* toc entry bullets */
        #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
            position: absolute;
            left: -15px;
            top: -1px;
            font-size: 1.5em;
        }

        /* toc entry, level 2 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
            content: "\2022";
        }

        /* toc entry, level 3 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
            content: "\25AB";
        }

        /* toc entry, level 4 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
            content: "-";
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* push <body> ("page") element down to make room for toc icon */
        .toc_body_nudge {
            padding-top: 60px;
        }

        /* toc icon when panel closed and not hovered */
        #toc_panel[data-open="false"]:not(:hover) {
            background: rgba(255, 255, 255, 0.75);
        }
    }

    /* always hide toc panel on print */
    @media only print {
        #toc_panel {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* lightbox plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* regular <img> in document when hovered */
        .lightbox_document_img:hover {
            cursor: pointer;
        }

        .body_no_scroll {
            overflow: hidden !important;
        }

        /* screen overlay */
        #lightbox_overlay {
            display: flex;
            flex-direction: column;
            position: fixed;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            background: rgba(0, 0, 0, 0.75);
            z-index: 3;
        }

        /* middle area containing lightbox image */
        #lightbox_image_container {
            flex-grow: 1;
            display: flex;
            justify-content: center;
            align-items: center;
            overflow: hidden;
            position: relative;
            padding: 20px;
        }

        /* bottom area containing caption */
        #lightbox_bottom_container {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 100px;
            min-height: 100px;
            max-height: 100px;
            background: rgba(0, 0, 0, 0.5);
        }

        /* image number info text box */
        #lightbox_number_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            left: 2px;
            top: 0;
            z-index: 4;
        }

        /* zoom info text box */
        #lightbox_zoom_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            right: 2px;
            top: 0;
            z-index: 4;
        }

        /* copy of image caption */
        #lightbox_caption {
            box-sizing: border-box;
            display: inline-block;
            width: 100%;
            max-height: 100%;
            padding: 10px 0;
            text-align: center;
            overflow-y: auto;
            color: #ffffff;
        }

        /* navigation previous/next button */
        .lightbox_button {
            width: 100px;
            height: 100%;
            min-width: 100px;
            min-height: 100%;
            color: #ffffff;
        }

        /* navigation previous/next button when hovered */
        .lightbox_button:hover {
            background: none !important;
        }

        /* navigation button icon */
        .lightbox_button > svg {
            height: 25px;
        }

        /* figure auto-number */
        #lightbox_caption > span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* lightbox image when hovered */
        #lightbox_img:hover {
            cursor: grab;
        }

        /* lightbox image when grabbed */
        #lightbox_img:active {
            cursor: grabbing;
        }
    }

    /* when on screen < 480px wide */
    @media only screen and (max-width: 480px) {
        /* make navigation buttons skinnier on small screens to make more room for caption text */
        .lightbox_button {
            width: 50px;
            min-width: 50px;
        }
    }

    /* always hide lightbox on print */
    @media only print {
        #lightbox_overlay {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* hypothesis (annotations) plugin */
    /* -------------------------------------------------- */

    /* side panel */
    .annotator-frame {
        width: 280px !important;
        z-index: 0 !important;
    }

    /* match highlight color to rest of theme */
    .annotator-highlights-always-on .annotator-hl {
        background-color: #ffeb3b !important;
    }

    /* match focused color to rest of theme */
    .annotator-hl.annotator-hl-focused {
        background-color: #ff8a65 !important;
    }

    /* match bucket bar color to rest of theme */
    .annotator-bucket-bar {
        background: #f5f5f5 !important;
    }

    /* always hide toolbar and tooltip on print */
    @media only print {
        .annotator-frame {
            display: none !important;
        }

        hypothesis-adder {
            display: none !important;
        }
    }
</style>
<!-- anchors plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds an anchor next to each of a certain type
        // of element that provides a human-readable url to that specific
        // item/position in the document (eg "manuscript.html#abstract"). It
        // also makes it such that scrolling out of view of a target removes
        // its identifier from the url.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'anchors';

        // default plugin options
        const options = {
            // which types of elements to add anchors next to, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3, [id^="fig:"], [id^="tbl:"], [id^="eq:"]',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // add anchor to each element of specified types
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements)
                addAnchor(element);

            // attach scroll listener to window
            window.addEventListener('scroll', onScroll);
        }

        // when window is scrolled
        function onScroll() {
            // if url has hash and user has scrolled out of view of hash
            // target, remove hash from url
            const tolerance = 100;
            const target = getHashTarget();
            if (target) {
                if (
                    target.getBoundingClientRect().top >
                        window.innerHeight + tolerance ||
                    target.getBoundingClientRect().bottom < 0 - tolerance
                )
                    history.pushState(null, null, ' ');
            }
        }

        // add anchor to element
        function addAnchor(element) {
            let addTo; // element to add anchor button to

            // if figure or table, modify withId and addTo to get expected
            // elements
            if (element.id.indexOf('fig:') === 0) {
                addTo = element.querySelector('figcaption');
            } else if (element.id.indexOf('tbl:') === 0) {
                addTo = element.querySelector('caption');
            } else if (element.id.indexOf('eq:') === 0) {
                addTo = element.querySelector('.eqnos-number');
            }

            addTo = addTo || element;
            const id = element.id || null;

            // do not add anchor if element doesn't have assigned id.
            // id is generated by pandoc and is assumed to be unique and
            // human-readable
            if (!id)
                return;

            // create anchor button
            const anchor = document.createElement('a');
            anchor.innerHTML = document.querySelector('.icon_link').innerHTML;
            anchor.title = 'Link to this part of the document';
            anchor.classList.add('icon_button', 'anchor');
            anchor.dataset.ignore = 'true';
            anchor.href = '#' + id;
            addTo.appendChild(anchor);
        }

        // get element that is target of link or url hash
        function getHashTarget() {
            const hash = window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- link icon -->

<template class="icon_link">
    <!-- modified from: https://fontawesome.com/icons/link -->
    <svg width="16" height="16" viewBox="0 0 512 512">
        <path
            fill="currentColor"
            d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        ></path>
    </svg>
</template>
<!-- accordion plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows sections of content under <h2> headings
        // to be collapsible.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'accordion';

        // default plugin options
        const options = {
            // whether to always start expanded ('false'), always start
            // collapsed ('true'), or start collapsed when screen small ('auto')
            startCollapsed: 'auto',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <h2> heading
            const headings = document.querySelectorAll('h2');
            for (const heading of headings) {
                addArrow(heading);

                // start expanded/collapsed based on option
                if (
                    options.startCollapsed === 'true' ||
                    (options.startCollapsed === 'auto' && isSmallScreen())
                )
                    collapseHeading(heading);
                else
                    expandHeading(heading);
            }

            // attach hash change listener to window
            window.addEventListener('hashchange', onHashChange);
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                goToElement(target);
        }

        // add arrow to heading
        function addArrow(heading) {
            // add arrow button
            const arrow = document.createElement('button');
            arrow.innerHTML = document.querySelector(
                '.icon_angle_down'
            ).innerHTML;
            arrow.classList.add('icon_button', 'accordion_arrow');
            heading.insertBefore(arrow, heading.firstChild);

            // attach click listener to heading and button
            heading.addEventListener('click', onHeadingClick);
            arrow.addEventListener('click', onArrowClick);
        }

        // determine if on mobile-like device with small screen
        function isSmallScreen() {
            return Math.min(window.innerWidth, window.innerHeight) < 480;
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get element that is target of hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // when <h2> heading is clicked
        function onHeadingClick(event) {
            // only collapse if <h2> itself is target of click (eg, user did
            // not click on anchor within <h2>)
            if (event.target === this)
                toggleCollapse(this);
        }

        // when arrow button is clicked
        function onArrowClick() {
            toggleCollapse(this.parentNode);
        }

        // collapse section if expanded, expand if collapsed
        function toggleCollapse(heading) {
            if (heading.dataset.collapsed === 'false')
                collapseHeading(heading);
            else
                expandHeading(heading);
        }

        // elements to exclude from collapse, such as table of contents panel,
        // hypothesis panel, etc
        const exclude = '#toc_panel, div.annotator-frame, #lightbox_overlay';

        // collapse section
        function collapseHeading(heading) {
            heading.setAttribute('data-collapsed', 'true');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'true');
        }

        // expand section
        function expandHeading(heading) {
            heading.setAttribute('data-collapsed', 'false');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'false');
        }

        // get list of elements between this <h2> and next <h2> or <h1>
        // ("children" of the <h2> section)
        function getChildren(heading) {
            return nextUntil(heading, 'h2, h1', exclude);
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get list of elements after a start element up to element matching
        // query
        function nextUntil(element, query, exclude) {
            const elements = [];
            while (element = element.nextElementSibling, element) {
                if (element.matches(query))
                    break;
                if (!element.matches(exclude))
                    elements.push(element);
            }
            return elements;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
    <!-- modified from: https://fontawesome.com/icons/angle-down -->
    <svg width="16" height="16" viewBox="0 0 448 512">
        <path
            fill="currentColor"
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        ></path>
    </svg>
</template>
<!-- tooltips plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when the user hovers or
        // focuses a link to a citation or figure, a tooltip appears with a
        // preview of the reference content, along with arrows to navigate
        // between instances of the same reference in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tooltips';

        // default plugin options
        const options = {
            // whether user must click off to close tooltip instead of just
            // un-hovering
            clickClose: 'false',
            // delay (in ms) between opening and closing tooltip
            delay: '100',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach hover and focus listeners to link
                link.addEventListener('mouseover', onLinkHover);
                link.addEventListener('mouseleave', onLinkUnhover);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('touchend', onLinkTouch);
            }

            // attach mouse, key, and resize listeners to window
            window.addEventListener('mousedown', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('keyup', onKeyUp);
            window.addEventListener('resize', onResize);
        }

        // when link is hovered
        function onLinkHover() {
            // function to open tooltip
            const delayOpenTooltip = function() {
                openTooltip(this);
            }.bind(this);

            // run open function after delay
            this.openTooltipTimer = window.setTimeout(
                delayOpenTooltip,
                options.delay
            );
        }

        // when mouse leaves link
        function onLinkUnhover() {
            // cancel opening tooltip
            window.clearTimeout(this.openTooltipTimer);

            // don't close on unhover if option specifies
            if (options.clickClose === 'true')
                return;

            // function to close tooltip
            const delayCloseTooltip = function() {
                // if tooltip open and if mouse isn't over tooltip, close
                const tooltip = document.getElementById('tooltip');
                if (tooltip && !tooltip.matches(':hover'))
                    closeTooltip();
            };

            // run close function after delay
            this.closeTooltipTimer = window.setTimeout(
                delayCloseTooltip,
                options.delay
            );
        }

        // when link is focused (tabbed to)
        function onLinkFocus(event) {
            openTooltip(this);
        }

        // when link is touched on touch screen
        function onLinkTouch(event) {
            // attempt to force hover state on first tap always, and trigger
            // regular link click (and navigation) on second tap
            if (event.target === document.activeElement)
                event.target.click();
            else {
                document.activeElement.blur();
                event.target.focus();
            }
            if (event.cancelable)
                event.preventDefault();
            event.stopPropagation();
            return false;
        }

        // when mouse is clicked anywhere in window
        function onClick(event) {
            closeTooltip();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'tooltip_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'tooltip_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeTooltip();
                    break;
            }
        }

        // when window is resized or zoomed
        function onResize() {
            closeTooltip();
        }

        // get all links of types we wish to handle
        function getLinks() {
            const queries = [];
            // exclude buttons, anchor links, toc links, etc
            const exclude =
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            queries.push('a[href^="#ref-"]' + exclude); // citation links
            queries.push('a[href^="#fig:"]' + exclude); // figure links
            const query = queries.join(', ');
            return document.querySelectorAll(query);
        }

        // get links with same target, get index of link in set, get total
        // same links
        function getSameLinks(link) {
            const sameLinks = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    sameLinks.push(otherLink);
            }

            return {
                elements: sameLinks,
                index: sameLinks.indexOf(link),
                total: sameLinks.length
            };
        }

        // open tooltip
        function openTooltip(link) {
            // delete tooltip if it exists, start fresh
            closeTooltip();

            // make tooltip element
            const tooltip = makeTooltip(link);

            // if source couldn't be found and tooltip not made, exit
            if (!tooltip)
                return;

            // make navbar elements
            const navBar = makeNavBar(link);
            if (navBar)
                tooltip.firstElementChild.appendChild(navBar);

            // attach tooltip to page
            document.body.appendChild(tooltip);

            // position tooltip
            const position = function() {
                positionTooltip(link);
            };
            position();

            // if tooltip contains images, position again after they've loaded
            const imgs = tooltip.querySelectorAll('img');
            for (const img of imgs)
                img.addEventListener('load', position);
        }

        // close (delete) tooltip
        function closeTooltip() {
            const tooltip = document.getElementById('tooltip');
            if (tooltip)
                tooltip.remove();
        }

        // make tooltip
        function makeTooltip(link) {
            // get target element that link points to
            const source = getSource(link);

            // if source can't be found, exit
            if (!source)
                return;

            // create new tooltip
            const tooltip = document.createElement('div');
            tooltip.id = 'tooltip';
            const tooltipContent = document.createElement('div');
            tooltipContent.id = 'tooltip_content';
            tooltip.appendChild(tooltipContent);

            // make copy of source node and put in tooltip
            const sourceCopy = makeCopy(source);
            tooltipContent.appendChild(sourceCopy);

            // attach mouse event listeners
            tooltip.addEventListener('click', onTooltipClick);
            tooltip.addEventListener('mousedown', onTooltipClick);
            tooltip.addEventListener('touchstart', onTooltipClick);
            tooltip.addEventListener('mouseleave', onTooltipUnhover);

            // (for interaction with lightbox plugin)
            // transfer click on tooltip copied img to original img
            const sourceImg = source.querySelector('img');
            const sourceCopyImg = sourceCopy.querySelector('img');
            if (sourceImg && sourceCopyImg) {
                const clickImg = function() {
                    sourceImg.click();
                    closeTooltip();
                };
                sourceCopyImg.addEventListener('click', clickImg);
            }

            return tooltip;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // when tooltip is clicked
        function onTooltipClick(event) {
            // when user clicks on tooltip, stop click from transferring
            // outside of tooltip (eg, click off to close tooltip, or eg click
            // off to unhighlight same refs)
            event.stopPropagation();
        }

        // when tooltip is unhovered
        function onTooltipUnhover(event) {
            if (options.clickClose === 'true')
                return;

            // make sure new mouse/touch/focus no longer over tooltip or any
            // element within it
            const tooltip = document.getElementById('tooltip');
            if (!tooltip)
                return;
            if (this.contains(event.relatedTarget))
                return;

            closeTooltip();
        }

        // make nav bar to go betwen prev/next instances of same reference
        function makeNavBar(link) {
            // find other links to the same source
            const sameLinks = getSameLinks(link);

            // don't show nav bar when singular reference
            if (sameLinks.total <= 1)
                return;

            // find prev/next links with same target
            const prevLink = getPrevLink(link, sameLinks);
            const nextLink = getNextLink(link, sameLinks);

            // create nav bar
            const navBar = document.createElement('div');
            navBar.id = 'tooltip_nav_bar';
            const text = sameLinks.index + 1 + ' of ' + sameLinks.total;

            // create nav bar prev/next buttons
            const prevButton = document.createElement('button');
            const nextButton = document.createElement('button');
            prevButton.id = 'tooltip_prev_button';
            nextButton.id = 'tooltip_next_button';
            prevButton.title =
                'Jump to the previous occurence of this item in the document [←]';
            nextButton.title =
                'Jump to the next occurence of this item in the document [→]';
            prevButton.classList.add('icon_button');
            nextButton.classList.add('icon_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;
            navBar.appendChild(prevButton);
            navBar.appendChild(document.createTextNode(text));
            navBar.appendChild(nextButton);

            // attach click listeners to buttons
            prevButton.addEventListener('click', function() {
                onPrevNextClick(link, prevLink);
            });
            nextButton.addEventListener('click', function() {
                onPrevNextClick(link, nextLink);
            });

            return navBar;
        }

        // get previous link with same target
        function getPrevLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if < 1
            let index;
            if (sameLinks.index - 1 >= 0)
                index = sameLinks.index - 1;
            else
                index = sameLinks.total - 1;
            return sameLinks.elements[index];
        }

        // get next link with same target
        function getNextLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if > total
            let index;
            if (sameLinks.index + 1 <= sameLinks.total - 1)
                index = sameLinks.index + 1;
            else
                index = 0;
            return sameLinks.elements[index];
        }

        // get element that is target of link or url hash
        function getSource(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if ref or figure, modify target to get expected element
            if (id.indexOf('ref-') === 0)
                target = target.querySelector('p');
            else if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');

            return target;
        }

        // when prev/next arrow button is clicked
        function onPrevNextClick(link, prevNextLink) {
            if (link && prevNextLink)
                goToElement(prevNextLink, window.innerHeight * 0.5);
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // determine position to place tooltip based on link position in
        // viewport and tooltip size
        function positionTooltip(link, left, top) {
            const tooltipElement = document.getElementById('tooltip');
            if (!tooltipElement)
                return;

            // get convenient vars for position/dimensions of
            // link/tooltip/page/view
            link = getRectInPage(link);
            const tooltip = getRectInPage(tooltipElement);
            const view = getRectInPage();

            // horizontal positioning
            if (left)
                // use explicit value
                left = left;
            else if (link.left + tooltip.width < view.right)
                // fit tooltip to right of link
                left = link.left;
            else if (link.right - tooltip.width > view.left)
                // fit tooltip to left of link
                left = link.right - tooltip.width;
            // center tooltip in view
            else
                left = (view.right - view.left) / 2 - tooltip.width / 2;

            // vertical positioning
            if (top)
                // use explicit value
                top = top;
            else if (link.top - tooltip.height > view.top)
                // fit tooltip above link
                top = link.top - tooltip.height;
            else if (link.bottom + tooltip.height < view.bottom)
                // fit tooltip below link
                top = link.bottom;
            else {
                // center tooltip in view
                top = view.top + view.height / 2 - tooltip.height / 2;
                // nudge off of link to left/right if possible
                if (link.right + tooltip.width < view.right)
                    left = link.right;
                else if (link.left - tooltip.width > view.left)
                    left = link.left - tooltip.width;
            }

            tooltipElement.style.left = left + 'px';
            tooltipElement.style.top = top + 'px';
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get position of element relative to page
        function getRectInPage(element) {
            const rect = getRectInView(element);
            const body = getRectInView(document.body);

            const newRect = {};
            newRect.left = rect.left - body.left;
            newRect.top = rect.top - body.top;
            newRect.right = rect.right - body.left;
            newRect.bottom = rect.bottom - body.top;
            newRect.width = rect.width;
            newRect.height = rect.height;

            return newRect;
        }

        // (for interaction with accordion plugin)
        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // (for interaction with accordion plugin)
        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- jump to first plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds a button next to each reference entry,
        // figure, and table that jumps the page to the first occurrence of a
        // link to that item in the manuscript.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'jumpToFirst';

        // default plugin options
        const options = {
            // whether to add buttons next to reference entries
            references: 'true',
            // whether to add buttons next to figures
            figures: 'true',
            // whether to add buttons next to tables
            tables: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            if (options.references !== 'false')
                makeReferenceButtons();
            if (options.figures !== 'false')
                makeFigureButtons();
            if (options.tables !== 'false')
                makeTableButtons();
        }

        // when jump button clicked
        function onButtonClick() {
            const first = getFirstOccurrence(this.dataset.id);
            if (!first)
                return;

            // update url hash so navigating "back" in history will return
            // user to jump button
            window.location.hash = this.dataset.id;
            // scroll to link
            window.setTimeout(function() {
                goToElement(first, window.innerHeight * 0.5);
            }, 0);
        }

        // get first occurence of link to item in document
        function getFirstOccurrence(id) {
            let query = 'a';
            query += '[href="#' + id + '"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelector(query);
        }

        // add button next to each reference entry
        function makeReferenceButtons() {
            const references = document.querySelectorAll('div[id^="ref-"]');
            for (const reference of references) {
                // get reference id and element to add button to
                const id = reference.id;
                const container = reference.firstElementChild;
                const first = getFirstOccurrence(id);

                // if can't find link to reference, ignore
                if (!first)
                    continue;

                // make jump button
                let button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this reference in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.innerHTML = button.outerHTML + container.innerHTML;
                button = container.firstElementChild;
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeFigureButtons() {
            const figures = document.querySelectorAll('[id^="fig:"]');
            for (const figure of figures) {
                // get figure id and element to add button to
                const id = figure.id;
                const container = figure.querySelector('figcaption') || figure;
                const first = getFirstOccurrence(id);

                // if can't find link to figure, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this figure in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeTableButtons() {
            const tables = document.querySelectorAll('[id^="tbl:"]');
            for (const table of tables) {
                // get ref id and element to add button to
                const id = table.id;
                const container = table.querySelector('caption') || table;
                const first = getFirstOccurrence(id);

                // if can't find link to table, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this table in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
    <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
    <svg width="16" height="16" viewBox="0 0 320 512">
        <path
            fill="currentColor"
            d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
        ></path>
    </svg>
</template>
<!-- link highlight plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user hovers or
        // focuses a link, other links that have the same target will be
        // highlighted. It also makes it such that when clicking a link, the
        // target of the link (eg reference, figure, table) is briefly
        // highlighted.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'linkHighlight';

        // default plugin options
        const options = {
            // whether to also highlight links that go to external urls
            externalLinks: 'false',
            // whether user must click off to unhighlight instead of just
            // un-hovering
            clickUnhighlight: 'false',
            // whether to also highlight links that are unique
            highlightUnique: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach mouse and focus listeners to link
                link.addEventListener('mouseenter', onLinkFocus);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('mouseleave', onLinkUnhover);
            }

            // attach click and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('hashchange', onHashChange);

            // run hash change on window load in case user has navigated
            // directly to hash
            onHashChange();
        }

        // when link is focused (tabbed to) or hovered
        function onLinkFocus() {
            highlight(this);
        }

        // when link is unhovered
        function onLinkUnhover() {
            if (options.clickUnhighlight !== 'true')
                unhighlightAll();
        }

        // when the mouse is clicked anywhere in window
        function onClick(event) {
            unhighlightAll();
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                glowElement(target);
        }

        // get element that is target of link or url hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            return target;
        }

        // start glow sequence on an element
        function glowElement(element) {
            const startGlow = function() {
                onGlowEnd();
                element.dataset.glow = 'true';
                element.addEventListener('animationend', onGlowEnd);
            };
            const onGlowEnd = function() {
                element.removeAttribute('data-glow');
                element.removeEventListener('animationend', onGlowEnd);
            };
            startGlow();
        }

        // highlight link and all others with same target
        function highlight(link) {
            // force unhighlight all to start fresh
            unhighlightAll();

            // get links with same target
            if (!link)
                return;
            const sameLinks = getSameLinks(link);

            // if link unique and option is off, exit and don't highlight
            if (sameLinks.length <= 1 && options.highlightUnique !== 'true')
                return;

            // highlight all same links, and "select" (special highlight) this
            // one
            for (const sameLink of sameLinks) {
                if (sameLink === link)
                    sameLink.setAttribute('data-selected', 'true');
                else
                    sameLink.setAttribute('data-highlighted', 'true');
            }
        }

        // unhighlight all links
        function unhighlightAll() {
            const links = getLinks();
            for (const link of links) {
                link.setAttribute('data-selected', 'false');
                link.setAttribute('data-highlighted', 'false');
            }
        }

        // get links with same target
        function getSameLinks(link) {
            const results = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    results.push(otherLink);
            }
            return results;
        }

        // get all links of types we wish to handle
        function getLinks() {
            let query = 'a';
            if (options.externalLinks !== 'true')
                query += '[href^="#"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelectorAll(query);
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- table of contents plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin provides a "table of contents" (toc) panel on
        // the side of the document that allows the user to conveniently
        // navigate between sections of the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tableOfContents';

        // default plugin options
        const options = {
            // which types of elements to add links for, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3',
            // whether default behavior is to be closed ('false'), open
            // ('true'), or only open when screen wide enough to fit panel
            // ('auto'). note: still always starts closed when page loads.
            open: 'auto',
            // if list item is more than this many characters, text will be
            // truncated
            charLimit: '50',
            // whether or not to show bullets next to each toc item
            bullets: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // make toc panel and populate with entries (links to document
            // sections)
            const panel = makePanel();
            if (!panel)
                return;
            makeEntries(panel);
            document.body.insertBefore(panel, document.body.firstChild);

            closePanel();

            // attach click, scroll, and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('scroll', onScroll);
            window.addEventListener('hashchange', onScroll);
            window.addEventListener('keyup', onKeyUp);
            onScroll();

            // add class to push document body down out of way of toc button
            document.body.classList.add('toc_body_nudge');
        }

        // determine if screen wide enough to fit toc panel
        function isSmallScreen() {
            // in default theme:
            // 816px = 8.5in = width of "page" (<body>) element
            // 260px = min width of toc panel (*2 for both sides of <body>)
            return window.innerWidth < 816 + 260 * 2;
        }

        // open/close panel based on option and screen size
        function openOrClosePanel() {
            if (
                options.open === 'true' ||
                (options.open === 'auto' && !isSmallScreen())
            )
                openPanel();
            else
                closePanel();
        }

        // when mouse is clicked anywhere in window
        function onClick() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                openOrClosePanel();
        }

        // when window is scrolled or hash changed
        function onScroll() {
            highlightViewed();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            // close on esc
            if (event.key === 'Escape')
                closePanel();
        }

        // find entry of currently viewed document section in toc and highlight
        function highlightViewed() {
            const firstId = getFirstInView(options.typesQuery);

            // get toc entries (links), unhighlight all, then highlight viewed
            const list = document.getElementById('toc_list');
            if (!firstId || !list)
                return;
            const links = list.querySelectorAll('a');
            for (const link of links)
                link.dataset.viewing = 'false';
            const link = list.querySelector('a[href="#' + firstId + '"]');
            if (!link)
                return;
            link.dataset.viewing = 'true';
        }

        // get first or previous toc listed element in top half of view
        function getFirstInView(query) {
            // get all elements matching query and with id
            const elements = document.querySelectorAll(query);
            const elementsWithIds = [];
            for (const element of elements) {
                if (element.id)
                    elementsWithIds.push(element);
            }


            // get first or previous element in top half of view
            for (let i = 0; i < elementsWithIds.length; i++) {
                const element = elementsWithIds[i];
                const prevElement = elementsWithIds[Math.max(0, i - 1)];
                if (element.getBoundingClientRect().top >= 0) {
                    if (
                        element.getBoundingClientRect().top <
                        window.innerHeight / 2
                    )
                        return element.id;
                    else
                        return prevElement.id;
                }
            }
        }

        // make panel
        function makePanel() {
            // create panel
            const panel = document.createElement('div');
            panel.id = 'toc_panel';
            if (options.bullets === 'true')
                panel.dataset.bullets = 'true';

            // create header
            const header = document.createElement('div');
            header.id = 'toc_header';

            // create toc button
            const button = document.createElement('button');
            button.id = 'toc_button';
            button.innerHTML = document.querySelector(
                '.icon_th_list'
            ).innerHTML;
            button.classList.add('icon_button');

            // create header text
            const text = document.createElement('h3');
            text.innerHTML = 'Table of Contents';

            // create container for toc list
            const list = document.createElement('div');
            list.id = 'toc_list';

            // attach click listeners
            panel.addEventListener('click', onPanelClick);
            header.addEventListener('click', onHeaderClick);
            button.addEventListener('click', onButtonClick);

            // attach elements
            header.appendChild(button);
            header.appendChild(text);
            panel.appendChild(header);
            panel.appendChild(list);

            return panel;
        }

        // create toc entries (links) to each element of the specified types
        function makeEntries(panel) {
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements) {
                // do not add link if element doesn't have assigned id
                if (!element.id)
                    continue;

                // create link/list item
                const link = document.createElement('a');
                link.classList.add('toc_link');
                switch (element.tagName.toLowerCase()) {
                    case 'h1':
                        link.dataset.level = '1';
                        break;
                    case 'h2':
                        link.dataset.level = '2';
                        break;
                    case 'h3':
                        link.dataset.level = '3';
                        break;
                    case 'h4':
                        link.dataset.level = '4';
                        break;
                }
                link.title = element.innerText;
                let text = element.innerText;
                if (text.length > options.charLimit)
                    text = text.slice(0, options.charLimit) + '...';
                link.innerHTML = text;
                link.href = '#' + element.id;
                link.addEventListener('click', onLinkClick);

                // attach link
                panel.querySelector('#toc_list').appendChild(link);
            }
        }

        // when panel is clicked
        function onPanelClick(event) {
            // stop click from propagating to window/document and closing panel
            event.stopPropagation();
        }

        // when header itself is clicked
        function onHeaderClick(event) {
            togglePanel();
        }

        // when button is clicked
        function onButtonClick(event) {
            togglePanel();
            // stop header underneath button from also being clicked
            event.stopPropagation();
        }

        // when link is clicked
        function onLinkClick() {
            openOrClosePanel();
        }

        // open panel if closed, close if opened
        function togglePanel() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                closePanel();
            else
                openPanel();
        }

        // open panel
        function openPanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'true';
        }

        // close panel
        function closePanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'false';
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- th list icon -->

<template class="icon_th_list">
    <!-- modified from: https://fontawesome.com/icons/th-list -->
    <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
        <path
            fill="currentColor"
            d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- lightbox plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user clicks on an
        // image, the image fills the screen and the user can pan/drag/zoom
        // the image and navigate between other images in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'lightbox';

        // default plugin options
        const options = {
            // list of possible zoom/scale factors
            zoomSteps:
                '0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1,' +
                '1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8',
            // whether to fit image to view ('fit'), display at 100% and shrink
            // if necessary ('shrink'), or always display at 100% ('100')
            defaultZoom: 'fit',
            // whether to zoom in/out toward center of view ('true') or mouse
            // ('false')
            centerZoom: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <img> element
            const imgs = document.querySelectorAll('figure > img');
            let count = 1;
            for (const img of imgs) {
                img.classList.add('lightbox_document_img');
                img.dataset.number = count;
                img.dataset.total = imgs.length;
                img.addEventListener('click', openLightbox);
                count++;
            }

            // attach mouse and key listeners to window
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('keyup', onKeyUp);
        }

        // when mouse is moved anywhere in window
        function onWindowMouseMove(event) {
            window.mouseX = event.clientX;
            window.mouseY = event.clientY;
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'lightbox_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'lightbox_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeLightbox();
                    break;
            }
        }

        // open lightbox
        function openLightbox() {
            const lightbox = makeLightbox(this);
            if (!lightbox)
                return;

            blurBody(lightbox);
            document.body.appendChild(lightbox);
        }

        // make lightbox
        function makeLightbox(img) {
            // delete lightbox if it exists, start fresh
            closeLightbox();

            // create screen overlay containing lightbox
            const overlay = document.createElement('div');
            overlay.id = 'lightbox_overlay';

            // create image info boxes
            const numberInfo = document.createElement('div');
            const zoomInfo = document.createElement('div');
            numberInfo.id = 'lightbox_number_info';
            zoomInfo.id = 'lightbox_zoom_info';

            // create container for image
            const imageContainer = document.createElement('div');
            imageContainer.id = 'lightbox_image_container';
            const lightboxImg = makeLightboxImg(
                img,
                imageContainer,
                numberInfo,
                zoomInfo
            );
            imageContainer.appendChild(lightboxImg);

            // create bottom container for caption and navigation buttons
            const bottomContainer = document.createElement('div');
            bottomContainer.id = 'lightbox_bottom_container';
            const caption = makeCaption(img);
            const prevButton = makePrevButton(img);
            const nextButton = makeNextButton(img);
            bottomContainer.appendChild(prevButton);
            bottomContainer.appendChild(caption);
            bottomContainer.appendChild(nextButton);

            // attach top middle and bottom to overlay
            overlay.appendChild(numberInfo);
            overlay.appendChild(zoomInfo);
            overlay.appendChild(imageContainer);
            overlay.appendChild(bottomContainer);

            return overlay;
        }

        // make <img> object that is intuitively draggable and zoomable
        function makeLightboxImg(
            sourceImg,
            container,
            numberInfoBox,
            zoomInfoBox
        ) {
            // create copy of source <img>
            const img = sourceImg.cloneNode(true);
            img.classList.remove('lightbox_document_img');
            img.removeAttribute('id');
            img.removeAttribute('width');
            img.removeAttribute('height');
            img.style.position = 'unset';
            img.style.margin = '0';
            img.style.padding = '0';
            img.style.width = '';
            img.style.height = '';
            img.style.minWidth = '';
            img.style.minHeight = '';
            img.style.maxWidth = '';
            img.style.maxHeight = '';
            img.id = 'lightbox_img';

            // build sorted list of unique zoomSteps, always including a 100%
            let zoomSteps = [];
            const optionsZooms = options.zoomSteps.split(/[^0-9.]/);
            for (const optionZoom of optionsZooms) {
                const newZoom = parseFloat(optionZoom);
                if (newZoom && !zoomSteps.includes(newZoom))
                    zoomSteps.push(newZoom);
            }
            if (!zoomSteps.includes(1))
                zoomSteps.push(1);
            zoomSteps = zoomSteps.sort(function sortNumber(a, b) {
                return a - b;
            });

            // <img> object property variables
            let zoom = 1;
            let translateX = 0;
            let translateY = 0;
            let clickMouseX = undefined;
            let clickMouseY = undefined;
            let clickTranslateX = undefined;
            let clickTranslateY = undefined;

            updateNumberInfo();

            // update image numbers displayed in info box
            function updateNumberInfo() {
                numberInfoBox.innerHTML =
                    sourceImg.dataset.number + ' of ' + sourceImg.dataset.total;
            }

            // update zoom displayed in info box
            function updateZoomInfo() {
                let zoomInfo = zoom * 100;
                if (!Number.isInteger(zoomInfo))
                    zoomInfo = zoomInfo.toFixed(2);
                zoomInfoBox.innerHTML = zoomInfo + '%';
            }

            // move to closest zoom step above current zoom
            const zoomIn = function() {
                for (const zoomStep of zoomSteps) {
                    if (zoomStep > zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                updateTransform();
            };

            // move to closest zoom step above current zoom
            const zoomOut = function() {
                zoomSteps.reverse();
                for (const zoomStep of zoomSteps) {
                    if (zoomStep < zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                zoomSteps.reverse();

                updateTransform();
            };

            // update display of <img> based on scale/translate properties
            const updateTransform = function() {
                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                // get new width/height after scale
                const rect = img.getBoundingClientRect();
                // limit translate
                translateX = Math.max(translateX, -rect.width / 2);
                translateX = Math.min(translateX, rect.width / 2);
                translateY = Math.max(translateY, -rect.height / 2);
                translateY = Math.min(translateY, rect.height / 2);

                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                updateZoomInfo();
            };

            // fit <img> to container
            const fit = function() {
                // no x/y offset, 100% zoom by default
                translateX = 0;
                translateY = 0;
                zoom = 1;

                // widths of <img> and container
                const imgWidth = img.naturalWidth;
                const imgHeight = img.naturalHeight;
                const containerWidth = parseFloat(
                    window.getComputedStyle(container).width
                );
                const containerHeight = parseFloat(
                    window.getComputedStyle(container).height
                );

                // how much zooming is needed to fit <img> to container
                const xRatio = imgWidth / containerWidth;
                const yRatio = imgHeight / containerHeight;
                const maxRatio = Math.max(xRatio, yRatio);
                const newZoom = 1 / maxRatio;

                // fit <img> to container according to option
                if (options.defaultZoom === 'shrink') {
                    if (maxRatio > 1)
                        zoom = newZoom;
                } else if (options.defaultZoom === 'fit')
                    zoom = newZoom;

                updateTransform();
            };

            // when mouse wheel is rolled anywhere in container
            const onContainerWheel = function(event) {
                if (!event)
                    return;

                // let ctrl + mouse wheel to zoom behave as normal
                if (event.ctrlKey)
                    return;

                // prevent normal scroll behavior
                event.preventDefault();
                event.stopPropagation();

                // point around which to scale img
                const viewRect = container.getBoundingClientRect();
                const viewX = (viewRect.left + viewRect.right) / 2;
                const viewY = (viewRect.top + viewRect.bottom) / 2;
                const originX = options.centerZoom === 'true' ? viewX : mouseX;
                const originY = options.centerZoom === 'true' ? viewY : mouseY;

                // get point on image under origin
                const oldRect = img.getBoundingClientRect();
                const oldPercentX = (originX - oldRect.left) / oldRect.width;
                const oldPercentY = (originY - oldRect.top) / oldRect.height;

                // increment/decrement zoom
                if (event.deltaY < 0)
                    zoomIn();
                if (event.deltaY > 0)
                    zoomOut();

                // get offset between previous image point and origin
                const newRect = img.getBoundingClientRect();
                const offsetX =
                    originX - (newRect.left + newRect.width * oldPercentX);
                const offsetY =
                    originY - (newRect.top + newRect.height * oldPercentY);

                // translate image to keep image point under origin
                translateX += offsetX;
                translateY += offsetY;

                // perform translate
                updateTransform();
            };

            // when container is clicked
            function onContainerClick(event) {
                // if container itself is target of click, and not other
                // element above it
                if (event.target === this)
                    closeLightbox();
            }

            // when mouse button is pressed on image
            const onImageMouseDown = function(event) {
                // store original mouse position relative to image
                clickMouseX = window.mouseX;
                clickMouseY = window.mouseY;
                clickTranslateX = translateX;
                clickTranslateY = translateY;
                event.stopPropagation();
                event.preventDefault();
            };

            // when mouse button is released anywhere in window
            const onWindowMouseUp = function(event) {
                // reset original mouse position
                clickMouseX = undefined;
                clickMouseY = undefined;
                clickTranslateX = undefined;
                clickTranslateY = undefined;

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mouseup', onWindowMouseUp);
            };

            // when mouse is moved anywhere in window
            const onWindowMouseMove = function(event) {
                if (
                    clickMouseX === undefined ||
                    clickMouseY === undefined ||
                    clickTranslateX === undefined ||
                    clickTranslateY === undefined
                )
                    return;

                // offset image based on original and current mouse position
                translateX = clickTranslateX + window.mouseX - clickMouseX;
                translateY = clickTranslateY + window.mouseY - clickMouseY;
                updateTransform();
                event.preventDefault();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mousemove', onWindowMouseMove);
            };

            // when window is resized
            const onWindowResize = function(event) {
                fit();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('resize', onWindowResize);
            };

            // attach the necessary event listeners
            img.addEventListener('dblclick', fit);
            img.addEventListener('mousedown', onImageMouseDown);
            container.addEventListener('wheel', onContainerWheel);
            container.addEventListener('mousedown', onContainerClick);
            container.addEventListener('touchstart', onContainerClick);
            window.addEventListener('mouseup', onWindowMouseUp);
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('resize', onWindowResize);

            // run fit() after lightbox atttached to document and <img> Loaded
            // so needed container and img dimensions available
            img.addEventListener('load', fit);

            return img;
        }

        // make caption
        function makeCaption(img) {
            const caption = document.createElement('div');
            caption.id = 'lightbox_caption';
            const captionSource = img.nextElementSibling;
            if (captionSource.tagName.toLowerCase() === 'figcaption') {
                const captionCopy = makeCopy(captionSource);
                caption.innerHTML = captionCopy.innerHTML;
            }

            caption.addEventListener('touchstart', function(event) {
                event.stopPropagation();
            });

            return caption;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // make button to jump to previous image in document
        function makePrevButton(img) {
            const prevButton = document.createElement('button');
            prevButton.id = 'lightbox_prev_button';
            prevButton.title = 'Jump to the previous image in the document [←]';
            prevButton.classList.add('icon_button', 'lightbox_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;

            // attach click listeners to button
            prevButton.addEventListener('click', function() {
                getPrevImg(img).click();
            });

            return prevButton;
        }

        // make button to jump to next image in document
        function makeNextButton(img) {
            const nextButton = document.createElement('button');
            nextButton.id = 'lightbox_next_button';
            nextButton.title = 'Jump to the next image in the document [→]';
            nextButton.classList.add('icon_button', 'lightbox_button');
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;

            // attach click listeners to button
            nextButton.addEventListener('click', function() {
                getNextImg(img).click();
            });

            return nextButton;
        }

        // get previous image in document
        function getPrevImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if < 1
            if (index - 1 >= 0)
                index--;
            else
                index = imgs.length - 1;
            return imgs[index];
        }

        // get next image in document
        function getNextImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if > total
            if (index + 1 <= imgs.length - 1)
                index++;
            else
                index = 0;
            return imgs[index];
        }

        // close lightbox
        function closeLightbox() {
            focusBody();

            const lightbox = document.getElementById('lightbox_overlay');
            if (lightbox)
                lightbox.remove();
        }

        // make all elements behind lightbox non-focusable
        function blurBody(overlay) {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.tabIndex = -1;
            document.body.classList.add('body_no_scroll');
        }

        // make all elements focusable again
        function focusBody() {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.removeAttribute('tabIndex');
            document.body.classList.remove('body_no_scroll');
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- attributes plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows arbitrary HTML attributes to be attached
        // to (almost) any element. Place an HTML comment inside or next to the
        // desired element in the format <!-- $attribute="value" -->

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'attributes';

        // default plugin options
        const options = {
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // get list of comments in document
            const comments = findComments();

            for(const comment of comments)
                if (comment.parentElement)
                    addAttributes(
                        comment.parentElement,
                        comment.nodeValue.trim()
                    );
        }

        // add html attributes to specified element based on string of 
        // html attributes and values
        function addAttributes(element, text) {
            // regex's for finding attribute/value pairs in the format of
            // attribute="value" or attribute='value
            const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
            const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

            // loop through attribute/value pairs
            let match;
            while(match = text.match(regex2) || text.match(regex1)) {
                // get attribute and value from regex capture groups
                let attribute = match[1];
                let value = match[2];

                // remove from string
                text = text.substring(match.index + match[0].length);

                if (!attribute || !value)
                    break;

                // set attribute of parent element
                try {
                    element.setAttribute(attribute, value);
                } catch(error) {
                    console.log(error);
                }

                // special case for colspan
                if (attribute === 'colspan')
                    removeTableCells(element, value);
            }
        }

        // get list of comment elements in document
        function findComments() {
            const comments = [];

            // iterate over comment nodes in document
            function acceptNode(node) {
                return NodeFilter.FILTER_ACCEPT;
            }
            const iterator = document.createNodeIterator(
                document.body,
                NodeFilter.SHOW_COMMENT,
                acceptNode
            );
            let node;
            while(node = iterator.nextNode())
                comments.push(node);

            return comments;
        }

        // remove certain number of cells after specified cell
        function removeTableCells(cell, number) {
            number = parseInt(number);
            if (!number)
                return;

            // remove elements
            for(; number > 1; number--) {
                if (cell.nextElementSibling)
                    cell.nextElementSibling.remove();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- math plugin configuration -->

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        "CommonHTML": { linebreaks: { automatic: true } },
        "HTML-CSS": { linebreaks: { automatic: true } },
        "SVG": { linebreaks: { automatic: true } },
        "fast-preview": { disabled: true }
    });
</script>

<!-- math plugin -->

<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'MathJax' allows the proper rendering of
    // math/equations written in LaTeX.

    // https://www.mathjax.org/
</script>
<!-- annotations plugin configuration -->

<script>
    window.hypothesisConfig = function() {
        return {
            branding: {
                accentColor: '#2196f3',
                appBackgroundColor: '#f8f8f8',
                ctaBackgroundColor: '#f8f8f8',
                ctaTextColor: '#000000',
                selectionFontFamily: 'Open Sans, Helvetica, sans serif',
                annotationFontFamily: 'Open Sans, Helvetica, sans serif'
            }
        };
    };
</script>

<!-- annotations plugin -->

<script src='https://hypothes.is/embed.js'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'Hypothesis' allows public annotation of the
    // manuscript.

    // https://web.hypothes.is/
</script>
<!-- analytics plugin -->

<!-- copy and paste code from Google Analytics or similar service here -->
</body>
</html>
